{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Use Amaranth 3RAD data to design sequence capture baits for a RADcap project.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Rerunning MrBait with the newest version of ipyrad (0.9.54) <br>\n",
    "June 15, 2020"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ipyrad 0.9.54\n"
     ]
    }
   ],
   "source": [
    "# import packages into python\n",
    "import ipyrad as ip\n",
    "import ipyrad.analysis as ipa\n",
    "\n",
    "# print the version of ipyrad you are running\n",
    "print('ipyrad', ipa.__version__)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### This is how I pooled together the reads from each species"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/sh\n",
    "#SBATCH --account=eaton\n",
    "# SBATCH --exclusive\n",
    "#SBATCH --time=11:59:00\n",
    "#SBATCH --workdir=/moto/eaton/users/sandra/\n",
    "#SBATCH --job-name=cat\n",
    "#SBATCH --mail-type=END\n",
    "#SBATCH --mail-user=sandra.hoffberg@columbia.edu\n",
    "\n",
    "\n",
    "mkdir /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "mkdir /moto/eaton/users/sandra/Just_pooled_species_fixed/\n",
    "\n",
    "cd /moto/eaton/users/sandra/cat_runs_after_i7/demux_inner_stacks/decloned/rename/\n",
    "\n",
    "cat SLH_AL_0001_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/acanthochiton_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0003_R1_.fq.gz SLH_AL_0004_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/acutilobus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0006_R1_.fq.gz SLH_AL_0009_R1_.fq.gz SLH_AL_0010_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/albus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0013_R1_.fq.gz SLH_AL_0018_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/arenicola_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0019_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/asplundii_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0020_R1_.fq.gz SLH_AL_0021_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/australis_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0023_R1_.fq.gz SLH_AL_0028_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/blitoides_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0029_R1_.fq.gz SLH_AL_0034_R1_.fq.gz SLH_AL_0037_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/blitum_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0039_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/californicus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0040_R1_.fq.gz SLH_AL_0041_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/cannabinus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0102_R1_.fq.gz SLH_AL_0110_R1_.fq.gz SLH_AL_0116_R1_.fq.gz SLH_AL_0322_R1_.fq.gz SLH_AL_0540_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/caudatus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0599_R1_.fq.gz SLH_AL_0600_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/crassipes_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0679_R1_.fq.gz SLH_AL_0699_R1_.fq.gz SLH_AL_0728_R1_.fq.gz SLH_AL_0804_R1_.fq.gz SLH_AL_0832_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/cruentus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0951_R1_.fq.gz SLH_AL_0952_R1_.fq.gz SLH_AL_0955_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/deflexus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0965_R1_.fq.gz SLH_AL_0979_R1_.fq.gz SLH_AL_0992_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/dubius_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0997_R1_.fq.gz SLH_AL_0998_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/fimbriatus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_1000_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/floridanus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_1009_R1_.fq.gz SLH_AL_1013_R1_.fq.gz SLH_AL_1014_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/graecizans_pooled_R1_.fq.gz\n",
    "cat SLH_AL_1015_R1_.fq.gz SLH_AL_1016_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/greggii_pooled_R1_.fq.gz\n",
    "cat SLH_AL_1060_R1_.fq.gz SLH_AL_1098_R1_.fq.gz SLH_AL_1099_R1_.fq.gz SLH_AL_1117_R1_.fq.gz SLH_AL_0001-restricted_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/hybridus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_1178_R1_.fq.gz SLH_AL_1197_R1_.fq.gz SLH_AL_1264_R1_.fq.gz SLH_AL_1285_R1_.fq.gz SLH_AL_2282_R1_.fq.gz SLH_AL_2436_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/hypochondriacus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2634_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/muricatus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2637_R1_.fq.gz SLH_AL_2644_R1_.fq.gz SLH_AL_2647_R1_.fq.gz SLH_AL_2649_R1_.fq.gz SLH_AL_0017_R1_.fq.gz SLH_AL_2650_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/palmeri_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2653_R1_.fq.gz SLH_AL_2663_R1_.fq.gz SLH_AL_2665_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/powellii_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2671_R1_.fq.gz SLH_AL_2675_R1_.fq.gz SLH_AL_2753_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/quitensis_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2770_R1_.fq.gz SLH_AL_2773_R1_.fq.gz SLH_AL_2780_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/retroflexus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2792_R1_.fq.gz SLH_AL_2793_R1_.fq.gz SLH_AL_2806_R1_.fq.gz SLH_AL_2809_R1_.fq.gz SLH_AL_2811_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/spinosus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2815_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/standleyanus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2869_R1_.fq.gz SLH_AL_2940_R1_.fq.gz SLH_AL_2953_R1_.fq.gz SLH_AL_2978_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/tricolor_pooled_R1_.fq.gz\n",
    "cat SLH_AL_3003_R1_.fq.gz SLH_AL_3017_R1_.fq.gz SLH_AL_3027_R1_.fq.gz SLH_AL_3045_R1_.fq.gz SLH_AL_0009-restricted_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/tuberculatus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_3047_R1_.fq.gz SLH_AL_3062_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/viridis_pooled_R1_.fq.gz\n",
    "cat SLH_AL_3066_R1_.fq.gz SLH_AL_3067_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/wrightii_pooled_R1_.fq.gz\n",
    "cat SLH_AL_3065_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/watsonii_pooled_R1_.fq.gz\n",
    "cp SLH_AL_0088-contemp_R1_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0095-contemp_R1_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0026-contemp_R1_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0101-contemp_R1_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0024-contemp_R1_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "\n",
    "cat SLH_AL_0001_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/acanthochiton_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0003_R2_.fq.gz SLH_AL_0004_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/acutilobus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0006_R2_.fq.gz SLH_AL_0009_R2_.fq.gz SLH_AL_0010_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/albus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0013_R2_.fq.gz SLH_AL_0018_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/arenicola_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0019_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/asplundii_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0020_R2_.fq.gz SLH_AL_0021_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/australis_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0023_R2_.fq.gz SLH_AL_0028_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/blitoides_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0029_R2_.fq.gz SLH_AL_0034_R2_.fq.gz SLH_AL_0037_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/blitum_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0039_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/californicus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0040_R2_.fq.gz SLH_AL_0041_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/cannabinus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0102_R2_.fq.gz SLH_AL_0110_R2_.fq.gz SLH_AL_0116_R2_.fq.gz SLH_AL_0322_R2_.fq.gz SLH_AL_0540_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/caudatus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0599_R2_.fq.gz SLH_AL_0600_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/crassipes_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0679_R2_.fq.gz SLH_AL_0699_R2_.fq.gz SLH_AL_0728_R2_.fq.gz SLH_AL_0804_R2_.fq.gz SLH_AL_0832_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/cruentus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0951_R2_.fq.gz SLH_AL_0952_R2_.fq.gz SLH_AL_0955_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/deflexus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0965_R2_.fq.gz SLH_AL_0979_R2_.fq.gz SLH_AL_0992_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/dubius_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0997_R2_.fq.gz SLH_AL_0998_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/fimbriatus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_1000_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/floridanus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_1009_R2_.fq.gz SLH_AL_1013_R2_.fq.gz SLH_AL_1014_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/graecizans_pooled_R2_.fq.gz\n",
    "cat SLH_AL_1015_R2_.fq.gz SLH_AL_1016_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/greggii_pooled_R2_.fq.gz\n",
    "cat SLH_AL_1060_R2_.fq.gz SLH_AL_1098_R2_.fq.gz SLH_AL_1099_R2_.fq.gz SLH_AL_1117_R2_.fq.gz SLH_AL_0001-restricted_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/hybridus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_1178_R2_.fq.gz SLH_AL_1197_R2_.fq.gz SLH_AL_1264_R2_.fq.gz SLH_AL_1285_R2_.fq.gz SLH_AL_2282_R2_.fq.gz SLH_AL_2436_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/hypochondriacus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2634_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/muricatus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2637_R2_.fq.gz SLH_AL_2644_R2_.fq.gz SLH_AL_2647_R2_.fq.gz SLH_AL_2649_R2_.fq.gz SLH_AL_0017_R2_.fq.gz SLH_AL_2650_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/palmeri_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2653_R2_.fq.gz SLH_AL_2663_R2_.fq.gz SLH_AL_2665_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/powellii_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2671_R2_.fq.gz SLH_AL_2675_R2_.fq.gz SLH_AL_2753_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/quitensis_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2770_R2_.fq.gz SLH_AL_2773_R2_.fq.gz SLH_AL_2780_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/retroflexus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2792_R2_.fq.gz SLH_AL_2793_R2_.fq.gz SLH_AL_2806_R2_.fq.gz SLH_AL_2809_R2_.fq.gz SLH_AL_2811_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/spinosus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2815_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/standleyanus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2869_R2_.fq.gz SLH_AL_2940_R2_.fq.gz SLH_AL_2953_R2_.fq.gz SLH_AL_2978_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/tricolor_pooled_R2_.fq.gz\n",
    "cat SLH_AL_3003_R2_.fq.gz SLH_AL_3017_R2_.fq.gz SLH_AL_3027_R2_.fq.gz SLH_AL_3045_R2_.fq.gz SLH_AL_0009-restricted_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/tuberculatus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_3047_R2_.fq.gz SLH_AL_3062_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/viridis_pooled_R2_.fq.gz\n",
    "cat SLH_AL_3066_R2_.fq.gz SLH_AL_3067_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/wrightii_pooled_R2_.fq.gz\n",
    "cat SLH_AL_3065_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/watsonii_pooled_R2_.fq.gz\n",
    "cp SLH_AL_0088-contemp_R2_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0095-contemp_R2_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0026-contemp_R2_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0101-contemp_R2_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0024-contemp_R2_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "\n",
    "\n",
    "cp /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/* /moto/eaton/users/sandra/Just_pooled_species_fixed/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then delete the contemp samples from folder Just_pooled_species_fixed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Since the names are ok already, don't need this cell\n",
    "# load accession:species dataframe\n",
    "spnames = pd.read_csv(\"/moto/eaton/projects/RAW_DATA/Amaranthus/Amaranth_sample_names.csv\")\n",
    "\n",
    "# subset only rows with species names not empty\n",
    "spnames = spnames[spnames.Species.notna()]\n",
    "spnames.Species = [i.split(\"-\")[0] for i in spnames.Species]\n",
    "\n",
    "# convert dataframe to a dictionary \n",
    "pdict = {spnames.Sample[i]: spnames.Species[i] for i in spnames.index}\n",
    "\n",
    "#create a merged assembly\n",
    "pooled = ip.merge(\n",
    "    name=\"pooled-spp33\", \n",
    "    assemblies=[lib1, lib2, lib3, lib4],\n",
    "    rename_dict=pdict,\n",
    "    )\n",
    "\n",
    "keep = [i for i in pooled.samples if \"SLH\" not in i]\n",
    "pooled = pooled.branch(\"pooled-spp33\", subsamples=keep, force=True)\n",
    "pooled.save()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New Assembly: pooled_9_45\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0   assembly_name               pooled_9_45                                  \n",
       "1   project_dir                 /moto/eaton/users/slh2181/MrBait             \n",
       "2   raw_fastq_path                                                           \n",
       "3   barcodes_path                                                            \n",
       "4   sorted_fastq_path           /moto/eaton/users/slh2181/Just_pooled_species_fixed/*\n",
       "5   assembly_method             reference                                    \n",
       "6   reference_sequence          /moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.hardmasked.fa\n",
       "7   datatype                    pair3rad                                     \n",
       "8   restriction_overhang        ('ATCGG', 'TAGCTT')                          \n",
       "9   max_low_qual_bases          6                                            \n",
       "10  phred_Qscore_offset         43                                           \n",
       "11  mindepth_statistical        6                                            \n",
       "12  mindepth_majrule            2                                            \n",
       "13  maxdepth                    10000                                        \n",
       "14  clust_threshold             0.85                                         \n",
       "15  max_barcode_mismatch        1                                            \n",
       "16  filter_adapters             3                                            \n",
       "17  filter_min_trim_len         35                                           \n",
       "18  max_alleles_consens         3                                            \n",
       "19  max_Ns_consens              0.2                                          \n",
       "20  max_Hs_consens              0.5                                          \n",
       "21  min_samples_locus           4                                            \n",
       "22  max_SNPs_locus              0.5                                          \n",
       "23  max_Indels_locus            12                                           \n",
       "24  max_shared_Hs_locus         0.4                                          \n",
       "25  trim_reads                  (0, 0, 0, 0)                                 \n",
       "26  trim_loci                   (0, 0, 0, 0)                                 \n",
       "27  output_formats              ['p', 's', 'l']                              \n",
       "28  pop_assign_file                                                          \n",
       "29  reference_as_filter                                                      "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HardMask = \"/moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.hardmasked.fa\"\n",
    "\n",
    "# set params on this assembly\n",
    "pooled = ip.Assembly(\"pooled_9_45\")\n",
    "pooled.params.sorted_fastq_path = \"/moto/eaton/users/slh2181/Just_pooled_species_fixed/*\"\n",
    "pooled.params.project_dir = \"/moto/eaton/users/slh2181/MrBait\"\n",
    "pooled.params.assembly_method = \"reference\"\n",
    "pooled.params.reference_sequence = HardMask\n",
    "pooled.params.restriction_overhang = (\"ATCGG\", \"TAGCTT\")\n",
    "pooled.params.datatype = \"pair3rad\"\n",
    "pooled.params.phred_Qscore_offset = 43\n",
    "pooled.params.max_barcode_mismatch = 1\n",
    "pooled.params.filter_adapters = 3\n",
    "pooled.params.max_alleles_consens = 3\n",
    "pooled.params.max_SNPs_locus = 0.5\n",
    "pooled.params.mindepth_majrule = 2\n",
    "pooled.params.mindepth_statistical = 6\n",
    "pooled.params.min_samples_locus = 4\n",
    "pooled.params.max_Hs_consens = 0.5\n",
    "pooled.params.max_Ns_consens = 0.2\n",
    "pooled.params.max_shared_Hs_locus = 0.4\n",
    "pooled.params.max_Indels_locus = 12\n",
    "pooled.params.output_formats = \"psl\"\n",
    "pooled.hackersonly.max_inner_mate_distance = 1000\n",
    "pooled.params.max_low_qual_bases = 6\n",
    "\n",
    "pooled.params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "! awk 1 /moto/eaton/users/slh2181/MrBait/hypochondriacus_cpDNA.fasta /moto/eaton/users/slh2181/MrBait/Ahypochondriacus_459_v2.0.hardmasked.fa > /moto/eaton/users/slh2181/MrBait/cp_nuclear_hardmask_genomes.fa\n",
    "\n",
    "! awk '!/^>/ {printf \"%s\", $0; n = \"\\n\"} /^>/ {print n $0; n = \"\"}' /moto/eaton/users/slh2181/MrBait/cp_nuclear_hardmask_genomes.fa | fold -w 60 > /moto/eaton/users/slh2181/MrBait/cp_nuclear_hardmask_genomes2.fa\n",
    "\n",
    "! echo \"\" >> /moto/eaton/users/slh2181/MrBait/cp_nuclear_hardmask_genomes2.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New Assembly: pooled_9_54\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0   assembly_name               pooled_9_54                                  \n",
       "1   project_dir                 /moto/eaton/users/slh2181/MrBait             \n",
       "2   raw_fastq_path                                                           \n",
       "3   barcodes_path                                                            \n",
       "4   sorted_fastq_path           /moto/eaton/users/slh2181/Just_pooled_species_fixed/*\n",
       "5   assembly_method             reference                                    \n",
       "6   reference_sequence          ./cp_nuclear_hardmask_genomes2.fa            \n",
       "7   datatype                    pair3rad                                     \n",
       "8   restriction_overhang        ('ATCGG', 'TAGCTT')                          \n",
       "9   max_low_qual_bases          6                                            \n",
       "10  phred_Qscore_offset         43                                           \n",
       "11  mindepth_statistical        6                                            \n",
       "12  mindepth_majrule            2                                            \n",
       "13  maxdepth                    10000                                        \n",
       "14  clust_threshold             0.85                                         \n",
       "15  max_barcode_mismatch        1                                            \n",
       "16  filter_adapters             3                                            \n",
       "17  filter_min_trim_len         35                                           \n",
       "18  max_alleles_consens         3                                            \n",
       "19  max_Ns_consens              0.2                                          \n",
       "20  max_Hs_consens              0.5                                          \n",
       "21  min_samples_locus           4                                            \n",
       "22  max_SNPs_locus              0.5                                          \n",
       "23  max_Indels_locus            12                                           \n",
       "24  max_shared_Hs_locus         0.4                                          \n",
       "25  trim_reads                  (0, 0, 0, 0)                                 \n",
       "26  trim_loci                   (0, 0, 0, 0)                                 \n",
       "27  output_formats              ['p', 's', 'l']                              \n",
       "28  pop_assign_file                                                          \n",
       "29  reference_as_filter                                                      "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cpHardMask = \"/moto/eaton/users/slh2181/MrBait/cp_nuclear_hardmask_genomes2.fa\"\n",
    "\n",
    "# set params on this assembly\n",
    "pooled = ip.Assembly(\"pooled_9_54\")\n",
    "pooled.params.sorted_fastq_path = \"/moto/eaton/users/slh2181/Just_pooled_species_fixed/*\"\n",
    "pooled.params.project_dir = \"/moto/eaton/users/slh2181/MrBait\"\n",
    "pooled.params.assembly_method = \"reference\"\n",
    "pooled.params.reference_sequence = cpHardMask\n",
    "pooled.params.restriction_overhang = (\"ATCGG\", \"TAGCTT\")\n",
    "pooled.params.datatype = \"pair3rad\"\n",
    "pooled.params.phred_Qscore_offset = 43\n",
    "pooled.params.max_barcode_mismatch = 1\n",
    "pooled.params.filter_adapters = 3\n",
    "pooled.params.max_alleles_consens = 3\n",
    "pooled.params.max_SNPs_locus = 0.5\n",
    "pooled.params.mindepth_majrule = 2\n",
    "pooled.params.mindepth_statistical = 6\n",
    "pooled.params.min_samples_locus = 4\n",
    "pooled.params.max_Hs_consens = 0.5\n",
    "pooled.params.max_Ns_consens = 0.2\n",
    "pooled.params.max_shared_Hs_locus = 0.4\n",
    "pooled.params.max_Indels_locus = 12\n",
    "pooled.params.output_formats = \"psl\"\n",
    "pooled.hackersonly.max_inner_mate_distance = 1000\n",
    "pooled.params.max_low_qual_bases = 6\n",
    "\n",
    "pooled.params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "loading Assembly: pooled_9_54\n",
      "from saved path: /moto/eaton/users/slh2181/MrBait/pooled_9_54.json\n"
     ]
    }
   ],
   "source": [
    "pooled = ip.load_json(\"/moto/eaton/users/slh2181/MrBait/pooled_9_54.json\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Parallel connection | t095: 24 cores\n",
      "[####################] 100% 0:00:29 | loading reads        | s1 |\n",
      "[####################] 100% 0:10:54 | processing reads     | s2 |\n",
      "[####################] 100% 0:05:28 | indexing reference   | s3 |\n",
      "[####################] 100% 0:01:14 | join unmerged pairs  | s3 |\n",
      "[####################] 100% 0:00:40 | dereplicating        | s3 |\n",
      "[####################] 100% 0:00:07 | splitting dereps     | s3 |\n",
      "[####################] 100% 0:06:48 | mapping reads        | s3 |\n",
      "[####################] 100% 2:40:04 | building clusters    | s3 |\n",
      "[####################] 100% 0:00:28 | calc cluster stats   | s3 |\n",
      "[                    ]   0% 0:01:07 | inferring [H, E]     | s4 |"
     ]
    }
   ],
   "source": [
    "# run assembly\n",
    "pooled.run(\"1234567\", force=True, auto=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "\t=======================================================================\r\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\r\n",
      "\t=======================================================================\r\n",
      "\r\n",
      "\tVersion:  1.2.6\r\n",
      "\tAuthor:   Tyler K. Chafin\r\n",
      "\tContact:  tkchafin@uark.edu\r\n",
      "\tLicense:  GNU Public License v3.0\r\n",
      "\r\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\r\n",
      "\r\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\r\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\r\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\r\n",
      "\r\n",
      "\tSource code: https://github.com/tkchafin/mrbait\r\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\r\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\r\n",
      "\r\n",
      "\t=======================================================================\r\n",
      "\t\r\n",
      "\r\n",
      "\tParsing command line arguments...\r\n",
      "\tExiting because help menu was called.\r\n",
      "\r\n",
      "MrBait Help Menu\r\n",
      "\r\n",
      "For more detailed information, please see the manual.\r\n",
      "\r\n",
      "\r\n",
      "Usage:\r\n",
      "\r\n",
      "\t /moto/home/slh2181/miniconda3/bin/mrbait -A </path/to/MAF> <-c><-l><-t><...>\r\n",
      "\r\n",
      "Description:\r\n",
      "\r\n",
      "\tMrBait is a program for designing baits for targeted enrichment studies\r\n",
      " \tincluding Ultraconverved Elements (UCEs), Anchored enrichment, and RAD-capture.\r\n",
      " \tIt can also be used to customize bait design for any of these methods based on\r\n",
      " \ta variety of built-in methods and constraints.\r\n",
      "\r\n",
      "\tNOTE: SQLite databases are currently remade from scratch with each\r\n",
      " \tinstance of the program. Support for opening and editing older database files\r\n",
      " \twill come soon.\r\n",
      "\r\n",
      "\r\n",
      "General options:\r\n",
      "\r\n",
      "\t-r,--resume\t: Resumes from a database pre-loaded with alignments (integer)\r\n",
      "\t\t--Options\r\n",
      "\t\t\t-r 1 : Continues pipeline beginning after Step 1 (Loading alignments)\r\n",
      "\t\t\t-r 2 : Continues after Step 2 (Target discovery)\r\n",
      "\t\t\t-r 3 : Continues after Step 3 (Target filtering/ selection)\r\n",
      "\t\t\t-r 4 : Continues after step 4 (Bait discovery)\r\n",
      "\t--db\t\t: .sqlite file containing pre-existing database. For use with --resume\r\n",
      "\t-T,--threads\t: Number of threads to use for processes that can run in parallel [1]\r\n",
      "\t-h,--help\t: Displays this help menu\r\n",
      "\t\r\n",
      "\r\n",
      "Input options:\r\n",
      "\r\n",
      "\t-M,--maf\t: Input multiple alignment MAF file\r\n",
      "\t-L,--loci\t: For RAD-data, as the \".loci\" output of pyRAD\r\n",
      "\t-A,--assembly\t: Input whole genome assembly as FASTA\r\n",
      "\t-X,--xmfa\t: Input whole genome alignments as XMFA\r\n",
      "\r\n",
      "Assembly input options (for use only with -A <genome.fasta>):\r\n",
      "\r\n",
      "\t-V,--vcf\t: VCF file containing variant information\r\n",
      "\t-G,--gff\t: GFF file containing annotation information\r\n",
      "\t-B,--bed\t: BED file containing regions (only 1 header line)\r\n",
      "\t--vcfALT\t: If using VCF and reference allele is gap/N, attempt\r\n",
      "\t\t \t   to call new consensus using VCF ALT alleles [default=OFF]\r\n",
      "\r\n",
      "Alignment filtering/ consensus options (use with -M or -L inputs):\r\n",
      "\r\n",
      "\t-c,--cov\t: Minimum number of sequences per alignment, for MAF or LOCI input [1]\r\n",
      "\t-l,--len\t: Minimum alignment length to attempt bait design [80]\r\n",
      "\t-q,--thresh\t: Threshold proportion for gap/N to include in consensus [0.1]\r\n",
      "\t-Q,--max_ambig\t: Maximum proportion of gap/N bases allowed in a consensus sequence [0.5]\r\n",
      "\t-k,--mask\t: Threshold proportion for masking (lower case) to include in consensus [0.1]\r\n",
      "\t-K,--max_mask\t: Maximum proportion of masked bases allowed in a consensus sequence [0.5]\r\n",
      "\t--consens_maf\t: Threshold minor allele frequency to code a SNP in consensus sequence [0.0]\r\n",
      "\t--dustMask\t: Use VSEARCH to mask consensus sequences using the DUST algorithm\r\n",
      "\r\n",
      "General Bait Design options:\r\n",
      "\r\n",
      "\t-b, --bait\t: Bait length [default=80]\r\n",
      "\t-w, --win_shift\t: Sliding window shift distance [1]\r\n",
      "\t\t  --Increasing will speed up program but could lower accuracy\r\n",
      "\t\t  --Cannot be longer than bait length <-b,--bait>\r\n",
      "\t-R,--mult_reg\t: Allow multiple target regions per locus [false]\r\n",
      "\t\t--See \"Target Region options\" below\r\n",
      "\t-m,--min_mult\t: Minimum alignment length to allow multiple regions\r\n",
      "\t\t--By default is set to the value of <-l>. Only applies when <-R>\r\n",
      "\t\t--When -R and -T are off, only one bait is selected per locus\r\n",
      "\t-v,--var_max\t: Maximum number of SNP columns to be allowed in a bait [0]\r\n",
      "\t\t--These will be coded as the appropriate IUPAC ambiguity code\r\n",
      "\t\t--Can be expanded in final output using the <--expand> flag\r\n",
      "\t-n,--numN\t: Number of consensus Ns allowed in a bait [0]\r\n",
      "\t\t--These will be inserted as \"N\" unless <-N> is used\r\n",
      "\t-g,--numG\t: Number of consensus indels (\"-\") allowed in bait [0]\r\n",
      "\r\n",
      "Target Region options:\r\n",
      "\r\n",
      "\t-D,--dist_r\t: Minimum distance between target regions [100]\r\n",
      "\t\t--Conflicts will be resolved according to --select_r\r\n",
      "\t-d,--flank_dist\t: Distance in flanking regions selection and filtering targets\r\n",
      "\t\t--See relevant options in <--select_r> and <--filter_r>\r\n",
      "\t\t--By default will be set to [500]\r\n",
      "\t\t--Appopriate value likely will correspond to your sequencing insert size\r\n",
      "\t--target_all\t: Skip target region search, and use full (passing) loci for bait design\r\n",
      "\t\t--Note that --select_r and --filter_r methods will still be applied\r\n",
      "\t-S, --select_r\t: Which criterion to select target regions w/in <-D>\r\n",
      "\t\t--Options\r\n",
      "\t\t\tsnp          : Most SNPs w/in \"d\" bases\r\n",
      "\t\t\tbad          : Least Ns and gaps w/n \"d\" bases\r\n",
      "\t\t\tcons         : Most conserved w/in \"d\" bases\r\n",
      "\t\t\trand         : Randomly choose a bait region [default]\r\n",
      "\t\t\tEx: -S snp -d 100 to choose region with most SNPs w/in 100 bases\r\n",
      "\t-F,--filter_r\t: Include any criteria used to filter ALL bait regions\r\n",
      "\t\t--Options\r\n",
      "\t\t\tlen=[x,y]    : Target length between \"x\" (min) and \"y\" (max)\r\n",
      "\t\t\tgap=[x]      : Maximum of \"x\" indels in target region\r\n",
      "\t\t\tbad=[x]      : Maximum of \"x\" Ns in target region\r\n",
      "\t\t\tsnp=[x,y]    : Min. of \"x\" and max \"y\" SNPs within --flank_dist\r\n",
      "\t\t\tmask=[x]     : Maximum of \"x\" proportion of masked bases allowed\r\n",
      "\t\t\tgc=[x,y]     : Proportion of G/C bases between \"x\" (min) and \"y\" (max)\r\n",
      "\t\t\trand=[x]     : Randomly retain \"x\" target regions w/ baits\r\n",
      "\t\t\tpw=[i,q]     : Pairwise alignment, removing when \"i\" identity in \"q\" proportion\r\n",
      "\t\t\tblast_i=[i,q]: Only retain hits over \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\tblast_x=[i,q]: Remove hits over \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\tblast_a=[i,q]: Remove targets having >1 hits with \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\tgff=[type]   : Only retain targets within \"d\" distance of \"type\" GFF elements\r\n",
      "\t\t\t               Use \"all\" to target all types, or see GFF3 specifications for SO feature types:\r\n",
      "\t\t\t               https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md\r\n",
      "\t\t\tgff_a=[alias]: Only retain targets within \"d\" distance of GFF records with attributes of \"Alias\"\r\n",
      "\t\t\t               Note that \"gff\" and \"gffa\" options are case-insensitive\r\n",
      "\t\t\tbed_x        : Exclude targets within \"d\" distance of BED records\r\n",
      "\t\t\tbed_i        : Only include targets within \"d\" distance of BED records\r\n",
      "\t\tEx1: -F snp=1,10 -d 100 to sample when 1-10 SNPs w/in 100 bases\r\n",
      "\t\tEx2: -F gc=0.2,0.8 -F rand=100 to randomly sample 100 targets with GC between 20-80%\r\n",
      "\t\tEx3: -F mask=0.0,0.1 to remove targets with >10% \\masked bases\r\n",
      "\t\tEx4: -F gff=exon -d 1000 to constrain targets to within 1000 bases of exons\r\n",
      "\t\tNOTE: Values for gc, mask, pw, and blast* methods are given as proportions (e.g. 90% = 0.9)\r\n",
      "\r\n",
      "Bait Design / Selection options:\r\n",
      "\r\n",
      "\t-s,--select_b\t: Which criterion to select a bait for target region\r\n",
      "\t\t--NOTE: This option is ignored when <-W>\r\n",
      "\t\t--Options\r\n",
      "\t\t\ttile=[x]     : Tile baits with \"x\" overlapping bases [default; x= bait_length/2]\r\n",
      "\t\t\tcenter=[n,x] : Design \"n\" centered baits with \"x\" overlap\r\n",
      "\t\t\tflank=[n,x]  : Design \"x\" terminal baits (each side) with \"x\" overlap\r\n",
      "\t\t\tcalc=[n,x]   : Design \"n\" baits per target, with \"x\" maximum overlap\r\n",
      "\t\t\tEx: -s tile=40 to tile baits with an overlap of 40 bases\r\n",
      "\t-f,--filter_b\t: Include any criteria used to filter ALL baits\r\n",
      "\t\t--Options\r\n",
      "\t\t\tmask=[x]     : Maximum of \"x\" proportion of masked bases allowed\r\n",
      "\t\t\tgc=[x,y]     : Proportion of G/C bases between \"x\" (min) and \"y\" (max)\r\n",
      "\t\t\tpw=[i,q]     : Pairwise alignment, removing when \"i\" identity in \"q\" proportion\r\n",
      "\t\t\trc=[i,q]     : Pairwise alignment against reverse-complements (see -r pw=[i,q])\r\n",
      "\t\t\tblast_i=[i,q]: Only retain hits over \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\tblast_x=[i,q]: Remove hits over \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\tblast_a=[i,q]: Remove targets having >1 hits with \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\trand=[x]     : Randomly retain \"x\" baits\r\n",
      "\t\t\tbed_x        : Exclude baits within \"d\" distance of BED records\r\n",
      "\t\t\tbed_i        : Only include baits within \"d\" distance of BED records\r\n",
      "\r\n",
      "VSEARCH Parameters (use when --select_b or --select_r = \"pw\" or \"rc\", or --dustMask):\r\n",
      "\r\n",
      "\t--vsearch\t: Path to VSEARCH executable if other than provided\r\n",
      "\t\t--MrBait will try to detect OS and appropriate exectable to use\r\n",
      "\t--vthreads\t: Number of threads for VSEARCH\r\n",
      "\t\t--Will set to the value of <--threads> if not defined\r\n",
      "\t--vsearch_qmask\t: Option for the --qmask option in VSEARCH\r\n",
      "\t\t--Only relevant when --select_b or --select_r = \"pw\" or \"rc\"\r\n",
      "\t\t--Options\r\n",
      "\t\t\tdust       : Mask sequences using the DUST algorithm [default]\r\n",
      "\t\t\tsoft       : Mask sequences using lower-cased characters\r\n",
      "\t\t\tnone       : Skip masking for pairwise alignments\r\n",
      "\r\n",
      "Graph Parameters (relevant for conflict resolution when --select_b or --select_r = \"pw\" or \"rc\"):\r\n",
      "\r\n",
      "\t--noGraph\t: Skip graph-based conflict resolution and remove ALL conflicting pairs\r\n",
      "\t--noWeightGraph\t: Unweighted (random) conflict resolution\r\n",
      "\t\t--MrBait will by default weight conflict resolution to maximize flanking SNPs\r\n",
      "\t--weightByMin\t: Weight loci by minimum ambiguities (NOT by flanking SNPs)\r\n",
      "\t--weightMax\t: Maximum graph size to attempt weighted algorithm\r\n",
      "\t\t--Default is 50,000. May be necessary for very large datasets with a lot of redundancy\r\n",
      "\r\n",
      "BLAST Parameters (use when --select_b or --select_r = \"blast\"):\r\n",
      "\r\n",
      "\t--blastdb\t: Path and prefix to existing Blast database to use\r\n",
      "\t--fastadb\t: Path and prefix to FASTA formatted sequences to make blast db\r\n",
      "\t--e_value\t: Minimum e-value cutoff for reporting BLAST hits [0.000001]\r\n",
      "\t--gapopen\t: Gap opening penalty for blastn [5]\r\n",
      "\t--gapextend\t: Gap extension penatly for blastn [2]\r\n",
      "\t--word_size\t: Word size for blastn\r\n",
      "\t--nodust\t: Turn off low-complexity filter for blastn\r\n",
      "\t--megablast\t: Use megablast rather than blastn\r\n",
      "\t\t--Default word size will be set to 28 if not provided\r\n",
      "\t--blastn\t: Path to blastn binary if different than default\r\n",
      "\t--makedb\t: Path to makeblastdb binary if different than default\r\n",
      "\t\t--MrBait will try to detect OS and appropriate exectable to use\r\n",
      "\t--max_hits\t: This value is passed directly to blast as --max_target_seqs\r\n",
      "\t\t--By default this is set at 10,000. Should only be necessary to change\r\n",
      "\t\t  if computation time becomes excessive with highly redundant databases\r\n",
      "\r\n",
      "Output options:\r\n",
      "\r\n",
      "\t-x,--expand\t: In output bait table, expand all ambiguities\r\n",
      "\t\t--Gaps are expanded as [ACGT] and absent\r\n",
      "\t\t--\"N\"s are expanded as [ACGT]\r\n",
      "\t--strand\t: Output strand. Options: \"+\", \"-\", \"both\"\r\n",
      "\t\t--\"-\" print antiparallel (reverse complement)\r\n",
      "\t-t,--print_tr\t: Boolean. Print target regions to FASTA file\r\n",
      "\t--print_loc\t: Boolean. Print consensus locus catalog to FASTA file\r\n",
      "\t-o,--out\t: Prefix for output files\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "! mrbait -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run MrBait with appropriate parameters (loose GC content)..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.2.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /moto/eaton/users/slh2181/MrBait/bait_nuc_and_cp_3jun2020.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:00 (0.023903 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum alignment depth (-c): 5\n",
      "\t\t\tMinimum alignment length (-l): 100\n",
      "\t\t\tThreshold to call \"N\" or gap consensus base (-q): 0.15\n",
      "\t\t\tThreshold to retain masked base (-k): 0.15\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.55\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.55\n",
      "\t\t\tMinor allele frequency threshold (--consens_maf): 0.2\n",
      "\t\tLoading LOCI file: /moto/eaton/users/slh2181/MrBait/pooled_9_45_outfiles/pooled_9_45.loci\n",
      "\t\t\tLoading alignments using 24 parallel processes.\n",
      "\t\t\tReading 36256 alignments... This may take a while.\n",
      "\t\tMasking consensus sequences using the DUST algorithm in VSEARCH... Done!\n",
      "\t\tFiltering loci... Done!\n",
      "\n",
      "\t\t### Results: 6871 loci passed filtering! ###\n",
      "\t\tPrinting locus catalog to file...\n",
      "\t\t### Runtime: 1:13:13 (4393.533810 seconds) ###\n",
      "\n",
      "\tStep 2: Target Discovery\n",
      "\t\tStep 2 parameters:\n",
      "\t\t\tSliding window width (-b, --bait): 90\n",
      "\t\t\tSliding window shift distance (-w, --win_shift): 1\n",
      "\t\t\tVariable columns allowed per target (-v, --var_max): 3\n",
      "\t\t\tAmbiguities (N's) allowed per target (-n, --numN): 1\n",
      "\t\t\tGap characters allowed per target (-g, --numG): 1\n",
      "\t\t\tFlanking distance to parse (-d,--flank_dist): 50\n",
      "\t\tStarting sliding window target discovery of 6871 loci...\n",
      "\t\t\tFinding targets using 24 parallel processes...\n",
      "\n",
      "\t\t### Results: 5458 potential targets identified! ###\n",
      "\t\t### Runtime: 0:00:24 (24.431269 seconds) ###\n",
      "\n",
      "\tStep 3: Target Filtering and Selection\n",
      "\t\tStep 3 parameters:\n",
      "\t\t\tMultiple targets allowed per locus (-R, --mult_reg): True\n",
      "\t\t\tMinimum locus length for multiple targets (-m, --min_mult): 200\n",
      "\t\t\tMinimum distance between targets within locus (-D, --dist_r): 30\n",
      "\t\t\tFlanking distance for target selection (-d,--flank_dist): 50\n",
      "\t\tApplying filters to 5458 targets.\n",
      "\t\t\tFiltering criterion: Between 1 and 15 flanking SNPs\n",
      "\t\t\tFiltering criterion: Between 0.2 and 0.8 GC content\n",
      "\t\t\tFiltering criterion: Target length between 125 and 9999\n",
      "\t\t\tFiltering criterion: BLAST inclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_nuc_and_cp_3jun2020\n",
      "\t\t\t  --Percent identity: 0.5\n",
      "\t\t\t  --Query coverage: 0.5\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 24\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\tResolving edges...\n",
      "\t\t\t  --Resolve by graph: True\n",
      "\t\t\t  --MIS method: Weighted\n",
      "\t\t\t  --Weights: Maximum variation\n",
      "\t\tResolving conflicting targets... Found 228 needing resolution.\n",
      "\t\t\tSelection criterion (-S, --select_r): Most conserved w/in 50 bases\n",
      "\t\t\tNote: Remaining conflicts will be randomly resolved.\n",
      "\t\tPrinting targets to file...\n",
      "\n",
      "\t\t### Results: 2062 targets passed filtering! ###\n",
      "\t\t### Runtime: 0:00:21 (21.048002 seconds) ###\n",
      "\n",
      "\tStep 4: Bait discovery\n",
      "\t\tStep 4 parameters:\n",
      "\t\t\tBait length (-b, --blen): 90\n",
      "\t\t\tTiling baits with 45 base overlap\n",
      "\n",
      "\t\t### Results: 4280 potential baits identified! ###\n",
      "\t\t### Runtime: 0:00:16 (16.746169 seconds) ###\n",
      "\n",
      "\tStep 5: Bait filtering\n",
      "\t\tFound 4280 potential baits!\n",
      "\t\tApplying filters to 4280 baits...\n",
      "\t\t\tFiltering criterion: Between 0.2 and 0.8 GC content\n",
      "\t\t\tFiltering criterion: BLAST exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_nuc_and_cp_3jun2020\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\t  --OVERRIDING BLAST DB: /moto/eaton/users/slh2181/reference/hypochondriacus_cpDNA.fasta\n",
      "\t\t\tFiltering criterion: BLAST ambiguous map exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_nuc_and_cp_3jun2020\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 24\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\tResolving edges...\n",
      "\t\t\t  --Resolve by graph: True\n",
      "\t\t\t  --MIS method: Weighted\n",
      "\t\t\t  --Weights: Maximum variation\n",
      "\t\t\t 4245 passed filtering.\n",
      "\t\tFormatting for printing...\n",
      "\t\tFinal baits output to: /moto/eaton/users/slh2181/MrBait/bait_nuc_and_cp_3jun2020.fasta\n",
      "\n",
      "\t\t### Results: 4245 baits output to file! ###\n",
      "\t\t### Runtime: 0:00:03 (3.598800 seconds) ###\n",
      "\n",
      "\t=======================================================================\n",
      "\t### Final output: /moto/eaton/users/slh2181/MrBait/bait_nuc_and_cp_3jun2020.fasta ###\n",
      "\t### Runtime: 1:14:19 (4459.383825 seconds) ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#nuclear DNA\n",
    "! mrbait -T 24 --vthreads 24 -L /moto/eaton/users/slh2181/MrBait/pooled_9_54_outfiles/pooled_9_54.loci --cov 5 -l 100 --thresh 0.15 -Q 0.55 --mask 0.15 --max_mask 0.55 -b 90 --var_max 3 --numN 1 --numG 1 --mult_reg --min_mult 200 --dist_r 30 --flank_dist 50 --select_r cons --filter_r snp=1,15 -s tile=45 --filter_r pw=0.8,0.8 --filter_b pw=0.8,0.8 --filter_r gc=0.2,0.8 --filter_b gc=0.2,0.8 --print_loc --print_tr --out bait_no_cp_3jun2020 --consens_maf 0.2 --filter_r len=125,9999 --dustMask --blastx_fdb /moto/eaton/users/slh2181/reference/hypochondriacus_cpDNA.fasta --filter_b blast_x=0.8,0.8 --fastadb /moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.fa --filter_r blast_i=0.5,0.5 --filter_b blast_a=0.8,0.8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.2.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /moto/eaton/users/slh2181/MrBait/bait_cp_3jun2020.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:51 (51.904741 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum alignment depth (-c): 2\n",
      "\t\t\tMinimum alignment length (-l): 100\n",
      "\t\t\tThreshold to call \"N\" or gap consensus base (-q): 0.15\n",
      "\t\t\tThreshold to retain masked base (-k): 0.15\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.55\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.55\n",
      "\t\t\tMinor allele frequency threshold (--consens_maf): 0.2\n",
      "\t\tLoading LOCI file: /moto/eaton/users/slh2181/MrBait/pooled_9_54_outfiles/pooled_9_54.loci\n",
      "\t\t\tLoading alignments using 24 parallel processes.\n",
      "Traceback (most recent call last):\n",
      "  File \"/moto/home/slh2181/miniconda3/bin/mrbait\", line 11, in <module>\n",
      "    load_entry_point('mrbait', 'console_scripts', 'mrbait')()\n",
      "  File \"/moto/home/slh2181/mrbait/mrbait/mrbait.py\", line 60, in main\n",
      "    loadAlignments(conn, params)\n",
      "  File \"/moto/home/slh2181/mrbait/mrbait/mrbait.py\", line 234, in loadAlignments\n",
      "    pcore.loadLOCI_parallel(conn, params)\n",
      "  File \"/moto/home/slh2181/mrbait/mrbait/mrbait_corefuncs_parallel.py\", line 232, in loadLOCI_parallel\n",
      "    numLoci = aln_file_tools.countLoci(params.loci)\n",
      "  File \"/moto/home/slh2181/mrbait/mrbait/aln_file_tools.py\", line 167, in countLoci\n",
      "    fh  = open(loci, 'r')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/moto/eaton/users/slh2181/MrBait/pooled_9_54_outfiles/pooled_9_54.loci'\n"
     ]
    }
   ],
   "source": [
    "#cpDNDA\n",
    "! mrbait -T 24 --vthreads 24 -L /moto/eaton/users/slh2181/MrBait/pooled_9_54_outfiles/pooled_9_54.loci --cov 2 -l 100 --thresh 0.15 -Q 0.55 --mask 0.15 --max_mask 0.55 -b 90 --var_max 3 --numN 1 --numG 1 --mult_reg --min_mult 200 --dist_r 30 --flank_dist 50 --select_r cons --filter_r snp=1,20 -s tile=45 --filter_r pw=0.8,0.8 --filter_b pw=0.8,0.8 --filter_r gc=0.2,0.8 --filter_b gc=0.2,0.8 --print_loc --print_tr --out bait_cp_3jun2020 --consens_maf 0.2 --filter_r len=100,9999 --dustMask --fastadb /moto/eaton/users/slh2181/reference/hypochondriacus_cpDNA.fasta --filter_b blast_x=0.8,0.8 --blastx_fdb /moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.fa --filter_r blast_i=0.4,0.4 --filter_b blast_a=0.6,0.6"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
