{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Use Amaranth 3RAD data to design sequence capture baits for a RADcap project.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Rerunning MrBait with the newest version of ipyrad (0.9.54) <br>\n",
    "June 15, 2020"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ipyrad 0.9.54\n"
     ]
    }
   ],
   "source": [
    "# import packages into python\n",
    "import ipyrad as ip\n",
    "import ipyrad.analysis as ipa\n",
    "\n",
    "# print the version of ipyrad you are running\n",
    "print('ipyrad', ipa.__version__)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### This is how I pooled together the reads from each species"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/sh\n",
    "#SBATCH --account=eaton\n",
    "# SBATCH --exclusive\n",
    "#SBATCH --time=11:59:00\n",
    "#SBATCH --workdir=/moto/eaton/users/sandra/\n",
    "#SBATCH --job-name=cat\n",
    "#SBATCH --mail-type=END\n",
    "#SBATCH --mail-user=sandra.hoffberg@columbia.edu\n",
    "\n",
    "\n",
    "mkdir /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "mkdir /moto/eaton/users/sandra/Just_pooled_species_fixed/\n",
    "\n",
    "cd /moto/eaton/users/sandra/cat_runs_after_i7/demux_inner_stacks/decloned/rename/\n",
    "\n",
    "cat SLH_AL_0001_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/acanthochiton_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0003_R1_.fq.gz SLH_AL_0004_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/acutilobus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0006_R1_.fq.gz SLH_AL_0009_R1_.fq.gz SLH_AL_0010_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/albus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0013_R1_.fq.gz SLH_AL_0018_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/arenicola_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0019_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/asplundii_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0020_R1_.fq.gz SLH_AL_0021_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/australis_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0023_R1_.fq.gz SLH_AL_0028_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/blitoides_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0029_R1_.fq.gz SLH_AL_0034_R1_.fq.gz SLH_AL_0037_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/blitum_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0039_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/californicus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0040_R1_.fq.gz SLH_AL_0041_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/cannabinus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0102_R1_.fq.gz SLH_AL_0110_R1_.fq.gz SLH_AL_0116_R1_.fq.gz SLH_AL_0322_R1_.fq.gz SLH_AL_0540_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/caudatus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0599_R1_.fq.gz SLH_AL_0600_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/crassipes_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0679_R1_.fq.gz SLH_AL_0699_R1_.fq.gz SLH_AL_0728_R1_.fq.gz SLH_AL_0804_R1_.fq.gz SLH_AL_0832_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/cruentus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0951_R1_.fq.gz SLH_AL_0952_R1_.fq.gz SLH_AL_0955_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/deflexus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0965_R1_.fq.gz SLH_AL_0979_R1_.fq.gz SLH_AL_0992_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/dubius_pooled_R1_.fq.gz\n",
    "cat SLH_AL_0997_R1_.fq.gz SLH_AL_0998_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/fimbriatus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_1000_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/floridanus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_1009_R1_.fq.gz SLH_AL_1013_R1_.fq.gz SLH_AL_1014_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/graecizans_pooled_R1_.fq.gz\n",
    "cat SLH_AL_1015_R1_.fq.gz SLH_AL_1016_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/greggii_pooled_R1_.fq.gz\n",
    "cat SLH_AL_1060_R1_.fq.gz SLH_AL_1098_R1_.fq.gz SLH_AL_1099_R1_.fq.gz SLH_AL_1117_R1_.fq.gz SLH_AL_0001-restricted_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/hybridus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_1178_R1_.fq.gz SLH_AL_1197_R1_.fq.gz SLH_AL_1264_R1_.fq.gz SLH_AL_1285_R1_.fq.gz SLH_AL_2282_R1_.fq.gz SLH_AL_2436_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/hypochondriacus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2634_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/muricatus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2637_R1_.fq.gz SLH_AL_2644_R1_.fq.gz SLH_AL_2647_R1_.fq.gz SLH_AL_2649_R1_.fq.gz SLH_AL_0017_R1_.fq.gz SLH_AL_2650_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/palmeri_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2653_R1_.fq.gz SLH_AL_2663_R1_.fq.gz SLH_AL_2665_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/powellii_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2671_R1_.fq.gz SLH_AL_2675_R1_.fq.gz SLH_AL_2753_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/quitensis_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2770_R1_.fq.gz SLH_AL_2773_R1_.fq.gz SLH_AL_2780_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/retroflexus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2792_R1_.fq.gz SLH_AL_2793_R1_.fq.gz SLH_AL_2806_R1_.fq.gz SLH_AL_2809_R1_.fq.gz SLH_AL_2811_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/spinosus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2815_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/standleyanus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_2869_R1_.fq.gz SLH_AL_2940_R1_.fq.gz SLH_AL_2953_R1_.fq.gz SLH_AL_2978_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/tricolor_pooled_R1_.fq.gz\n",
    "cat SLH_AL_3003_R1_.fq.gz SLH_AL_3017_R1_.fq.gz SLH_AL_3027_R1_.fq.gz SLH_AL_3045_R1_.fq.gz SLH_AL_0009-restricted_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/tuberculatus_pooled_R1_.fq.gz\n",
    "cat SLH_AL_3047_R1_.fq.gz SLH_AL_3062_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/viridis_pooled_R1_.fq.gz\n",
    "cat SLH_AL_3066_R1_.fq.gz SLH_AL_3067_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/wrightii_pooled_R1_.fq.gz\n",
    "cat SLH_AL_3065_R1_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/watsonii_pooled_R1_.fq.gz\n",
    "cp SLH_AL_0088-contemp_R1_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0095-contemp_R1_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0026-contemp_R1_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0101-contemp_R1_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0024-contemp_R1_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "\n",
    "cat SLH_AL_0001_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/acanthochiton_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0003_R2_.fq.gz SLH_AL_0004_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/acutilobus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0006_R2_.fq.gz SLH_AL_0009_R2_.fq.gz SLH_AL_0010_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/albus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0013_R2_.fq.gz SLH_AL_0018_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/arenicola_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0019_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/asplundii_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0020_R2_.fq.gz SLH_AL_0021_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/australis_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0023_R2_.fq.gz SLH_AL_0028_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/blitoides_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0029_R2_.fq.gz SLH_AL_0034_R2_.fq.gz SLH_AL_0037_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/blitum_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0039_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/californicus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0040_R2_.fq.gz SLH_AL_0041_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/cannabinus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0102_R2_.fq.gz SLH_AL_0110_R2_.fq.gz SLH_AL_0116_R2_.fq.gz SLH_AL_0322_R2_.fq.gz SLH_AL_0540_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/caudatus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0599_R2_.fq.gz SLH_AL_0600_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/crassipes_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0679_R2_.fq.gz SLH_AL_0699_R2_.fq.gz SLH_AL_0728_R2_.fq.gz SLH_AL_0804_R2_.fq.gz SLH_AL_0832_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/cruentus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0951_R2_.fq.gz SLH_AL_0952_R2_.fq.gz SLH_AL_0955_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/deflexus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0965_R2_.fq.gz SLH_AL_0979_R2_.fq.gz SLH_AL_0992_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/dubius_pooled_R2_.fq.gz\n",
    "cat SLH_AL_0997_R2_.fq.gz SLH_AL_0998_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/fimbriatus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_1000_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/floridanus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_1009_R2_.fq.gz SLH_AL_1013_R2_.fq.gz SLH_AL_1014_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/graecizans_pooled_R2_.fq.gz\n",
    "cat SLH_AL_1015_R2_.fq.gz SLH_AL_1016_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/greggii_pooled_R2_.fq.gz\n",
    "cat SLH_AL_1060_R2_.fq.gz SLH_AL_1098_R2_.fq.gz SLH_AL_1099_R2_.fq.gz SLH_AL_1117_R2_.fq.gz SLH_AL_0001-restricted_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/hybridus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_1178_R2_.fq.gz SLH_AL_1197_R2_.fq.gz SLH_AL_1264_R2_.fq.gz SLH_AL_1285_R2_.fq.gz SLH_AL_2282_R2_.fq.gz SLH_AL_2436_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/hypochondriacus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2634_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/muricatus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2637_R2_.fq.gz SLH_AL_2644_R2_.fq.gz SLH_AL_2647_R2_.fq.gz SLH_AL_2649_R2_.fq.gz SLH_AL_0017_R2_.fq.gz SLH_AL_2650_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/palmeri_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2653_R2_.fq.gz SLH_AL_2663_R2_.fq.gz SLH_AL_2665_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/powellii_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2671_R2_.fq.gz SLH_AL_2675_R2_.fq.gz SLH_AL_2753_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/quitensis_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2770_R2_.fq.gz SLH_AL_2773_R2_.fq.gz SLH_AL_2780_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/retroflexus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2792_R2_.fq.gz SLH_AL_2793_R2_.fq.gz SLH_AL_2806_R2_.fq.gz SLH_AL_2809_R2_.fq.gz SLH_AL_2811_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/spinosus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2815_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/standleyanus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_2869_R2_.fq.gz SLH_AL_2940_R2_.fq.gz SLH_AL_2953_R2_.fq.gz SLH_AL_2978_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/tricolor_pooled_R2_.fq.gz\n",
    "cat SLH_AL_3003_R2_.fq.gz SLH_AL_3017_R2_.fq.gz SLH_AL_3027_R2_.fq.gz SLH_AL_3045_R2_.fq.gz SLH_AL_0009-restricted_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/tuberculatus_pooled_R2_.fq.gz\n",
    "cat SLH_AL_3047_R2_.fq.gz SLH_AL_3062_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/viridis_pooled_R2_.fq.gz\n",
    "cat SLH_AL_3066_R2_.fq.gz SLH_AL_3067_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/wrightii_pooled_R2_.fq.gz\n",
    "cat SLH_AL_3065_R2_.fq.gz > /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/watsonii_pooled_R2_.fq.gz\n",
    "cp SLH_AL_0088-contemp_R2_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0095-contemp_R2_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0026-contemp_R2_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0101-contemp_R2_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "cp SLH_AL_0024-contemp_R2_.fq.gz /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/\n",
    "\n",
    "\n",
    "cp /moto/eaton/users/sandra/Pooled_species_Palmeri_fixed/* /moto/eaton/users/sandra/Just_pooled_species_fixed/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then delete the contemp samples from folder Just_pooled_species_fixed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Since the names are ok already, don't need this cell\n",
    "# load accession:species dataframe\n",
    "spnames = pd.read_csv(\"/moto/eaton/projects/RAW_DATA/Amaranthus/Amaranth_sample_names.csv\")\n",
    "\n",
    "# subset only rows with species names not empty\n",
    "spnames = spnames[spnames.Species.notna()]\n",
    "spnames.Species = [i.split(\"-\")[0] for i in spnames.Species]\n",
    "\n",
    "# convert dataframe to a dictionary \n",
    "pdict = {spnames.Sample[i]: spnames.Species[i] for i in spnames.index}\n",
    "\n",
    "#create a merged assembly\n",
    "pooled = ip.merge(\n",
    "    name=\"pooled-spp33\", \n",
    "    assemblies=[lib1, lib2, lib3, lib4],\n",
    "    rename_dict=pdict,\n",
    "    )\n",
    "\n",
    "keep = [i for i in pooled.samples if \"SLH\" not in i]\n",
    "pooled = pooled.branch(\"pooled-spp33\", subsamples=keep, force=True)\n",
    "pooled.save()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New Assembly: pooled_9_45\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0   assembly_name               pooled_9_45                                  \n",
       "1   project_dir                 /moto/eaton/users/slh2181/MrBait             \n",
       "2   raw_fastq_path                                                           \n",
       "3   barcodes_path                                                            \n",
       "4   sorted_fastq_path           /moto/eaton/users/slh2181/Just_pooled_species_fixed/*\n",
       "5   assembly_method             reference                                    \n",
       "6   reference_sequence          /moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.hardmasked.fa\n",
       "7   datatype                    pair3rad                                     \n",
       "8   restriction_overhang        ('ATCGG', 'TAGCTT')                          \n",
       "9   max_low_qual_bases          6                                            \n",
       "10  phred_Qscore_offset         43                                           \n",
       "11  mindepth_statistical        6                                            \n",
       "12  mindepth_majrule            2                                            \n",
       "13  maxdepth                    10000                                        \n",
       "14  clust_threshold             0.85                                         \n",
       "15  max_barcode_mismatch        1                                            \n",
       "16  filter_adapters             3                                            \n",
       "17  filter_min_trim_len         35                                           \n",
       "18  max_alleles_consens         3                                            \n",
       "19  max_Ns_consens              0.2                                          \n",
       "20  max_Hs_consens              0.5                                          \n",
       "21  min_samples_locus           4                                            \n",
       "22  max_SNPs_locus              0.5                                          \n",
       "23  max_Indels_locus            12                                           \n",
       "24  max_shared_Hs_locus         0.4                                          \n",
       "25  trim_reads                  (0, 0, 0, 0)                                 \n",
       "26  trim_loci                   (0, 0, 0, 0)                                 \n",
       "27  output_formats              ['p', 's', 'l']                              \n",
       "28  pop_assign_file                                                          \n",
       "29  reference_as_filter                                                      "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HardMask = \"/moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.hardmasked.fa\"\n",
    "\n",
    "# set params on this assembly\n",
    "pooled = ip.Assembly(\"pooled_9_45\")\n",
    "pooled.params.sorted_fastq_path = \"/moto/eaton/users/slh2181/Just_pooled_species_fixed/*\"\n",
    "pooled.params.project_dir = \"/moto/eaton/users/slh2181/MrBait\"\n",
    "pooled.params.assembly_method = \"reference\"\n",
    "pooled.params.reference_sequence = HardMask\n",
    "pooled.params.restriction_overhang = (\"ATCGG\", \"TAGCTT\")\n",
    "pooled.params.datatype = \"pair3rad\"\n",
    "pooled.params.phred_Qscore_offset = 43\n",
    "pooled.params.max_barcode_mismatch = 1\n",
    "pooled.params.filter_adapters = 3\n",
    "pooled.params.max_alleles_consens = 3\n",
    "pooled.params.max_SNPs_locus = 0.5\n",
    "pooled.params.mindepth_majrule = 2\n",
    "pooled.params.mindepth_statistical = 6\n",
    "pooled.params.min_samples_locus = 4\n",
    "pooled.params.max_Hs_consens = 0.5\n",
    "pooled.params.max_Ns_consens = 0.2\n",
    "pooled.params.max_shared_Hs_locus = 0.4\n",
    "pooled.params.max_Indels_locus = 12\n",
    "pooled.params.output_formats = \"psl\"\n",
    "pooled.hackersonly.max_inner_mate_distance = 1000\n",
    "pooled.params.max_low_qual_bases = 6\n",
    "\n",
    "pooled.params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "! awk 1 /home/sandra/Files/MrBait/hypochondriacus_cpDNA.fasta /home/sandra/Files/MrBait/Ahypochondriacus_459_v2.0.hardmasked.fa > /home/sandra/Files/MrBait/cp_nuclear_hardmask_genomes.fa\n",
    "\n",
    "! awk '!/^>/ {printf \"%s\", $0; n = \"\\n\"} /^>/ {print n $0; n = \"\"}' /home/sandra/Files/MrBait/cp_nuclear_hardmask_genomes.fa | fold -w 60 > /home/sandra/Files/MrBait/cp_nuclear_hardmask_genomes2.fa\n",
    "\n",
    "! echo \"\" >> /home/sandra/Files/MrBait/cp_nuclear_hardmask_genomes2.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New Assembly: pooled_9_54\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0   assembly_name               pooled_9_54                                  \n",
       "1   project_dir                 ~/Files/MrBait                               \n",
       "2   raw_fastq_path                                                           \n",
       "3   barcodes_path                                                            \n",
       "4   sorted_fastq_path           ~/Files/Just_pooled_species_fixed/*          \n",
       "5   assembly_method             reference                                    \n",
       "6   reference_sequence          ./cp_nuclear_hardmask_genomes2.fa            \n",
       "7   datatype                    pair3rad                                     \n",
       "8   restriction_overhang        ('ATCGG', 'TAGCTT')                          \n",
       "9   max_low_qual_bases          6                                            \n",
       "10  phred_Qscore_offset         43                                           \n",
       "11  mindepth_statistical        6                                            \n",
       "12  mindepth_majrule            2                                            \n",
       "13  maxdepth                    10000                                        \n",
       "14  clust_threshold             0.85                                         \n",
       "15  max_barcode_mismatch        1                                            \n",
       "16  filter_adapters             3                                            \n",
       "17  filter_min_trim_len         35                                           \n",
       "18  max_alleles_consens         3                                            \n",
       "19  max_Ns_consens              0.2                                          \n",
       "20  max_Hs_consens              0.5                                          \n",
       "21  min_samples_locus           4                                            \n",
       "22  max_SNPs_locus              0.5                                          \n",
       "23  max_Indels_locus            12                                           \n",
       "24  max_shared_Hs_locus         0.4                                          \n",
       "25  trim_reads                  (0, 0, 0, 0)                                 \n",
       "26  trim_loci                   (0, 0, 0, 0)                                 \n",
       "27  output_formats              ['p', 's', 'l']                              \n",
       "28  pop_assign_file                                                          \n",
       "29  reference_as_filter                                                      "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cpHardMask = \"/home/sandra/Files/MrBait/cp_nuclear_hardmask_genomes2.fa\"\n",
    "\n",
    "# set params on this assembly\n",
    "pooled = ip.Assembly(\"pooled_9_54\")\n",
    "pooled.params.sorted_fastq_path = \"/home/sandra/Files/Just_pooled_species_fixed/*\"\n",
    "pooled.params.project_dir = \"/home/sandra/Files/MrBait\"\n",
    "pooled.params.assembly_method = \"reference\"\n",
    "pooled.params.reference_sequence = cpHardMask\n",
    "pooled.params.restriction_overhang = (\"ATCGG\", \"TAGCTT\")\n",
    "pooled.params.datatype = \"pair3rad\"\n",
    "pooled.params.phred_Qscore_offset = 43\n",
    "pooled.params.max_barcode_mismatch = 1\n",
    "pooled.params.filter_adapters = 3\n",
    "pooled.params.max_alleles_consens = 3\n",
    "pooled.params.max_SNPs_locus = 0.5\n",
    "pooled.params.mindepth_majrule = 2\n",
    "pooled.params.mindepth_statistical = 6\n",
    "pooled.params.min_samples_locus = 4\n",
    "pooled.params.max_Hs_consens = 0.5\n",
    "pooled.params.max_Ns_consens = 0.2\n",
    "pooled.params.max_shared_Hs_locus = 0.4\n",
    "pooled.params.max_Indels_locus = 12\n",
    "pooled.params.output_formats = \"psl\"\n",
    "#pooled.hackersonly.max_inner_mate_distance = 1000\n",
    "pooled.params.max_low_qual_bases = 6\n",
    "\n",
    "pooled.params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Parallel connection | pinky: 80 cores\n",
      "[####################] 100% 0:00:30 | loading reads        | s1 |\n",
      "[####################] 100% 0:12:37 | processing reads     | s2 |\n",
      "[####################] 100% 0:05:30 | indexing reference   | s3 |\n",
      "[####################] 100% 0:01:18 | join unmerged pairs  | s3 |\n",
      "[####################] 100% 0:00:16 | dereplicating        | s3 |\n",
      "[####################] 100% 0:00:05 | splitting dereps     | s3 |\n",
      "[####################] 100% 0:08:22 | mapping reads        | s3 |\n",
      "[####################] 100% 0:49:55 | building clusters    | s3 |\n",
      "[####################] 100% 0:00:10 | calc cluster stats   | s3 |\n",
      "[####################] 100% 0:12:57 | inferring [H, E]     | s4 |\n",
      "[####################] 100% 0:00:09 | calculating depths   | s5 |\n",
      "[####################] 100% 0:00:17 | chunking clusters    | s5 |\n",
      "[####################] 100% 0:11:10 | consens calling      | s5 |\n",
      "[####################] 100% 0:01:44 | indexing alleles     | s5 |\n",
      "[####################] 100% 0:00:12 | concatenating bams   | s6 |\n",
      "[####################] 100% 0:00:02 | fetching regions     | s6 |\n",
      "[####################] 100% 0:00:04 | building database    | s6 |\n",
      "[####################] 100% 0:00:14 | applying filters     | s7 |\n",
      "[####################] 100% 0:01:33 | building arrays      | s7 |\n",
      "[####################] 100% 0:00:51 | writing conversions  | s7 |\n",
      "Parallel connection closed.\n"
     ]
    }
   ],
   "source": [
    "# run assembly\n",
    "pooled.run(\"1234567\", force=True, auto=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>reference</th>\n",
       "      <td>37316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acanthochiton_pooled</th>\n",
       "      <td>9782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acutilobus_pooled</th>\n",
       "      <td>16142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>albus_pooled</th>\n",
       "      <td>14578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arenicola_pooled</th>\n",
       "      <td>17120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tricolor_pooled</th>\n",
       "      <td>12541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tuberculatus_pooled</th>\n",
       "      <td>17567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>viridis_pooled</th>\n",
       "      <td>11861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>watsonii_pooled</th>\n",
       "      <td>9017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wrightii_pooled</th>\n",
       "      <td>14325</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>34 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sample_coverage\n",
       "reference                       37316\n",
       "acanthochiton_pooled             9782\n",
       "acutilobus_pooled               16142\n",
       "albus_pooled                    14578\n",
       "arenicola_pooled                17120\n",
       "...                               ...\n",
       "tricolor_pooled                 12541\n",
       "tuberculatus_pooled             17567\n",
       "viridis_pooled                  11861\n",
       "watsonii_pooled                  9017\n",
       "wrightii_pooled                 14325\n",
       "\n",
       "[34 rows x 1 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pooled.stats_dfs.s7_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### How many loci on each chromosome? "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold_name</th>\n",
       "      <th>scaffold_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KX279888_1_A_hypo_chloroplast_complete_genome</td>\n",
       "      <td>150518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Scaffold_10</td>\n",
       "      <td>22670516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Scaffold_11</td>\n",
       "      <td>22280117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Scaffold_12</td>\n",
       "      <td>22052327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Scaffold_13</td>\n",
       "      <td>20679869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Scaffold_14</td>\n",
       "      <td>20190685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Scaffold_15</td>\n",
       "      <td>17522127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Scaffold_16</td>\n",
       "      <td>16951160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Scaffold_1</td>\n",
       "      <td>38124660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Scaffold_2</td>\n",
       "      <td>35657244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Scaffold_3</td>\n",
       "      <td>30204323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Scaffold_4</td>\n",
       "      <td>28349311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Scaffold_5</td>\n",
       "      <td>25672467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Scaffold_6</td>\n",
       "      <td>24628041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Scaffold_7</td>\n",
       "      <td>24364990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Scaffold_8</td>\n",
       "      <td>23766980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Scaffold_9</td>\n",
       "      <td>22691259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Contig177|quiver|</td>\n",
       "      <td>1092290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Contig536|quiver|</td>\n",
       "      <td>146947</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Contig747|quiver|</td>\n",
       "      <td>124984</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    scaffold_name  scaffold_length\n",
       "0   KX279888_1_A_hypo_chloroplast_complete_genome           150518\n",
       "1                                     Scaffold_10         22670516\n",
       "2                                     Scaffold_11         22280117\n",
       "3                                     Scaffold_12         22052327\n",
       "4                                     Scaffold_13         20679869\n",
       "5                                     Scaffold_14         20190685\n",
       "6                                     Scaffold_15         17522127\n",
       "7                                     Scaffold_16         16951160\n",
       "8                                      Scaffold_1         38124660\n",
       "9                                      Scaffold_2         35657244\n",
       "10                                     Scaffold_3         30204323\n",
       "11                                     Scaffold_4         28349311\n",
       "12                                     Scaffold_5         25672467\n",
       "13                                     Scaffold_6         24628041\n",
       "14                                     Scaffold_7         24364990\n",
       "15                                     Scaffold_8         23766980\n",
       "16                                     Scaffold_9         22691259\n",
       "17                              Contig177|quiver|          1092290\n",
       "18                              Contig536|quiver|           146947\n",
       "19                              Contig747|quiver|           124984"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = \"/home/sandra/Files/MrBait/pooled_9_54_outfiles/pooled_9_54.seqs.hdf5\"\n",
    "chro = ipa.window_extracter(data)\n",
    "chro.scaffold_table.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wrote data to /home/sandra/Files/MrBait/cp.phy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>sites</th>\n",
       "      <th>snps</th>\n",
       "      <th>missing</th>\n",
       "      <th>samples</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>prefilter</th>\n",
       "      <td>KX279888_1_A_hypo_chloroplast_complete_genome</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>13318</td>\n",
       "      <td>102</td>\n",
       "      <td>0.71</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postfilter</th>\n",
       "      <td>KX279888_1_A_hypo_chloroplast_complete_genome</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>10228</td>\n",
       "      <td>98</td>\n",
       "      <td>0.65</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 scaffold  start   end  sites  \\\n",
       "prefilter   KX279888_1_A_hypo_chloroplast_complete_genome      0  None  13318   \n",
       "postfilter  KX279888_1_A_hypo_chloroplast_complete_genome      0  None  10228   \n",
       "\n",
       "            snps  missing  samples  \n",
       "prefilter    102     0.71       34  \n",
       "postfilter    98     0.65       34  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chro = ipa.window_extracter(data, \n",
    "                            name=\"cp\",\n",
    "                            workdir=\"/home/sandra/Files/MrBait\",\n",
    "                           scaffold_idxs=[0])\n",
    "chro.run()\n",
    "chro.stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "rax = ipa.raxml(data = \"/home/sandra/Files/MrBait/cp.phy\", name=\"cp_RAX\", workdir=\"/home/sandra/Files/MrBait\",T=80, N=100, m=\"GTRGAMMA\" )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "job cp_RAX finished successfully\n"
     ]
    }
   ],
   "source": [
    "rax.run()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(<toyplot.canvas.Canvas at 0x7ff84fa8fa90>,\n",
       " <toyplot.coordinates.Cartesian at 0x7ff84d1d57d0>)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div class=\"toyplot\" id=\"t165ed59554964f419b1ef2bb70fc18ce\" style=\"text-align:center\"><svg class=\"toyplot-canvas-Canvas\" height=\"612.0px\" id=\"tb36a481816bc4a8fa378ad649a925047\" preserveAspectRatio=\"xMidYMid meet\" style=\"background-color:transparent;border-color:#292724;border-style:none;border-width:1.0;fill:rgb(16.1%,15.3%,14.1%);fill-opacity:1.0;font-family:Helvetica;font-size:12px;opacity:1.0;stroke:rgb(16.1%,15.3%,14.1%);stroke-opacity:1.0;stroke-width:1.0\" viewBox=\"0 0 360.0 612.0\" width=\"360.0px\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:toyplot=\"http://www.sandia.gov/toyplot\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"><g class=\"toyplot-coordinates-Cartesian\" id=\"t3d7f08af014b4b479026cd56a7ba9cb3\"><clipPath id=\"ta9dcfadf4f024d21910dc13b4dc89925\"><rect height=\"552.0\" width=\"300.0\" x=\"30.0\" y=\"30.0\"></rect></clipPath><g clip-path=\"url(#ta9dcfadf4f024d21910dc13b4dc89925)\"><g class=\"toyplot-mark-Graph\" id=\"tb9a5a6944b6047f18faa6a077899d89a\"><g class=\"toyplot-Edges\"><path d=\"M 50.0 111.26298783724519 L 117.53246753246752 111.26298783724519\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 50.0 312.82406702208675 L 61.25541125541126 312.82406702208675\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 117.53246753246752 69.66867455976364 L 140.04329004329003 69.66867455976364\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 117.53246753246752 152.8573011147268 L 128.78787878787878 152.8573011147268\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 61.25541125541126 260.62438551547456 L 128.78787878787878 260.62438551547456\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 61.25541125541126 365.023748528699 L 72.5108225108225 365.023748528699\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 56.434120335110435 L 173.80952380952382 56.434120335110435\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 82.90322878441685 L 151.2987012987013 82.90322878441685\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 124.4975420618984 L 162.55411255411255 124.4975420618984\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 181.21706016755513 L 140.04329004329003 181.21706016755513\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 237.93657827321186 L 173.80952380952382 237.93657827321186\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 283.31219275773725 L 140.04329004329003 283.31219275773725\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 72.5108225108225 343.81301207043776 L 173.80952380952382 343.81301207043776\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 72.5108225108225 386.2344849869602 L 83.76623376623377 386.2344849869602\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 71.55932516328556 L 173.80952380952382 71.55932516328556\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 94.2471324055482 L 162.55411255411255 94.2471324055482\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 116.93493964781089 L 173.80952380952382 116.93493964781089\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 132.06014447598596 L 173.80952380952382 132.06014447598596\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 158.52925292529244 L 151.2987012987013 158.52925292529244\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 203.90486740981783 L 151.2987012987013 203.90486740981783\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 260.62438551547456 L 162.55411255411255 260.62438551547456\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 305.99999999999994 L 151.2987012987013 305.99999999999994\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 83.76623376623377 358.9382168986129 L 173.80952380952382 358.9382168986129\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 83.76623376623377 413.5307530753075 L 95.02164502164501 413.5307530753075\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 86.68452999146064 L 173.80952380952382 86.68452999146064\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 101.80973481963576 L 173.80952380952382 101.80973481963576\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 147.1853493041611 L 173.80952380952382 147.1853493041611\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 169.8731565464238 L 162.55411255411255 169.8731565464238\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 192.56096378868648 L 173.80952380952382 192.56096378868648\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 215.24877103094917 L 162.55411255411255 215.24877103094917\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 253.061783101387 L 173.80952380952382 253.061783101387\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 268.1869879295621 L 173.80952380952382 268.1869879295621\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 290.87479517182487 L 162.55411255411255 290.87479517182487\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 321.1252048281751 L 162.55411255411255 321.1252048281751\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 95.02164502164501 385.40732534791937 L 151.2987012987013 385.40732534791937\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 95.02164502164501 441.6541808026957 L 106.27705627705627 441.6541808026957\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 162.31055413233622 L 173.80952380952382 162.31055413233622\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 177.43575896051135 L 173.80952380952382 177.43575896051135\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 207.6861686168616 L 173.80952380952382 207.6861686168616\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 222.81137344503674 L 173.80952380952382 222.81137344503674\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 283.31219275773725 L 173.80952380952382 283.31219275773725\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 298.4373975859124 L 173.80952380952382 298.4373975859124\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 313.5626024140875 L 173.80952380952382 313.5626024140875\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 328.68780724226264 L 173.80952380952382 328.68780724226264\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 374.063421726788 L 173.80952380952382 374.063421726788\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 396.7512289690507 L 162.55411255411255 396.7512289690507\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 106.27705627705627 419.4390362113134 L 173.80952380952382 419.4390362113134\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 106.27705627705627 463.86932539407786 L 117.53246753246752 463.86932539407786\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 389.18862655496315 L 173.80952380952382 389.18862655496315\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 404.3138313831383 L 173.80952380952382 404.3138313831383\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 117.53246753246752 434.56424103948854 L 173.80952380952382 434.56424103948854\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 117.53246753246752 493.17440974866713 L 128.78787878787878 493.17440974866713\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 462.9240000923169 L 140.04329004329003 462.9240000923169\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 523.4248194050174 L 140.04329004329003 523.4248194050174\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 449.68944586766366 L 173.80952380952382 449.68944586766366\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 476.15855431697014 L 151.2987012987013 476.15855431697014\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 510.1902651803642 L 173.80952380952382 510.1902651803642\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 536.6593736296707 L 151.2987012987013 536.6593736296707\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 464.81465069583885 L 173.80952380952382 464.81465069583885\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 487.5024579381015 L 162.55411255411255 487.5024579381015\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 525.3154700085394 L 173.80952380952382 525.3154700085394\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 548.0032772508021 L 162.55411255411255 548.0032772508021\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 479.9398555240139 L 173.80952380952382 479.9398555240139\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 495.06506035218905 L 173.80952380952382 495.06506035218905\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 540.4406748367144 L 173.80952380952382 540.4406748367144\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 555.5658796648896 L 173.80952380952382 555.5658796648896\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 50.0 212.04352742966597 L 50.0 111.26298783724519\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 50.0 212.04352742966597 L 50.0 312.82406702208675\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 117.53246753246752 111.26298783724519 L 117.53246753246752 69.66867455976364\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 117.53246753246752 111.26298783724519 L 117.53246753246752 152.8573011147268\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 61.25541125541126 312.82406702208675 L 61.25541125541126 260.62438551547456\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 61.25541125541126 312.82406702208675 L 61.25541125541126 365.023748528699\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 69.66867455976364 L 140.04329004329003 56.434120335110435\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 69.66867455976364 L 140.04329004329003 82.90322878441685\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 152.8573011147268 L 128.78787878787878 124.4975420618984\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 152.8573011147268 L 128.78787878787878 181.21706016755513\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 260.62438551547456 L 128.78787878787878 237.93657827321186\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 260.62438551547456 L 128.78787878787878 283.31219275773725\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 72.5108225108225 365.023748528699 L 72.5108225108225 343.81301207043776\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 72.5108225108225 365.023748528699 L 72.5108225108225 386.2344849869602\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 82.90322878441685 L 151.2987012987013 71.55932516328556\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 82.90322878441685 L 151.2987012987013 94.2471324055482\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 124.4975420618984 L 162.55411255411255 116.93493964781089\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 124.4975420618984 L 162.55411255411255 132.06014447598596\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 181.21706016755513 L 140.04329004329003 158.52925292529244\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 181.21706016755513 L 140.04329004329003 203.90486740981783\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 283.31219275773725 L 140.04329004329003 260.62438551547456\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 283.31219275773725 L 140.04329004329003 305.99999999999994\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 83.76623376623377 386.2344849869602 L 83.76623376623377 358.9382168986129\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 83.76623376623377 386.2344849869602 L 83.76623376623377 413.5307530753075\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 94.2471324055482 L 162.55411255411255 86.68452999146064\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 94.2471324055482 L 162.55411255411255 101.80973481963576\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 158.52925292529244 L 151.2987012987013 147.1853493041611\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 158.52925292529244 L 151.2987012987013 169.8731565464238\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 203.90486740981783 L 151.2987012987013 192.56096378868648\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 203.90486740981783 L 151.2987012987013 215.24877103094917\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 260.62438551547456 L 162.55411255411255 253.061783101387\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 260.62438551547456 L 162.55411255411255 268.1869879295621\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 305.99999999999994 L 151.2987012987013 290.87479517182487\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 305.99999999999994 L 151.2987012987013 321.1252048281751\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 95.02164502164501 413.5307530753075 L 95.02164502164501 385.40732534791937\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 95.02164502164501 413.5307530753075 L 95.02164502164501 441.6541808026957\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 169.8731565464238 L 162.55411255411255 162.31055413233622\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 169.8731565464238 L 162.55411255411255 177.43575896051135\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 215.24877103094917 L 162.55411255411255 207.6861686168616\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 215.24877103094917 L 162.55411255411255 222.81137344503674\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 290.87479517182487 L 162.55411255411255 283.31219275773725\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 290.87479517182487 L 162.55411255411255 298.4373975859124\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 321.1252048281751 L 162.55411255411255 313.5626024140875\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 321.1252048281751 L 162.55411255411255 328.68780724226264\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 385.40732534791937 L 151.2987012987013 374.063421726788\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 385.40732534791937 L 151.2987012987013 396.7512289690507\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 106.27705627705627 441.6541808026957 L 106.27705627705627 419.4390362113134\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 106.27705627705627 441.6541808026957 L 106.27705627705627 463.86932539407786\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 396.7512289690507 L 162.55411255411255 389.18862655496315\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 396.7512289690507 L 162.55411255411255 404.3138313831383\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 117.53246753246752 463.86932539407786 L 117.53246753246752 434.56424103948854\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 117.53246753246752 463.86932539407786 L 117.53246753246752 493.17440974866713\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 493.17440974866713 L 128.78787878787878 462.9240000923169\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 128.78787878787878 493.17440974866713 L 128.78787878787878 523.4248194050174\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 462.9240000923169 L 140.04329004329003 449.68944586766366\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 462.9240000923169 L 140.04329004329003 476.15855431697014\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 523.4248194050174 L 140.04329004329003 510.1902651803642\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 140.04329004329003 523.4248194050174 L 140.04329004329003 536.6593736296707\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 476.15855431697014 L 151.2987012987013 464.81465069583885\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 476.15855431697014 L 151.2987012987013 487.5024579381015\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 536.6593736296707 L 151.2987012987013 525.3154700085394\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 151.2987012987013 536.6593736296707 L 151.2987012987013 548.0032772508021\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 487.5024579381015 L 162.55411255411255 479.9398555240139\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 487.5024579381015 L 162.55411255411255 495.06506035218905\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 548.0032772508021 L 162.55411255411255 540.4406748367144\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><path d=\"M 162.55411255411255 548.0032772508021 L 162.55411255411255 555.5658796648896\" style=\"fill:none;opacity:1;stroke:rgb(14.9%,14.9%,14.9%);stroke-linecap:round;stroke-opacity:1.0;stroke-width:2.0\"></path><g class=\"toyplot-HeadMarkers\"></g><g class=\"toyplot-MiddleMarkers\"></g><g class=\"toyplot-TailMarkers\"></g></g><g class=\"toyplot-Vertices\"><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 555.5658796648896)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 540.4406748367144)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 525.3154700085394)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 510.1902651803642)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 495.06506035218905)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 479.9398555240139)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 464.81465069583885)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 449.68944586766366)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 434.56424103948854)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 419.4390362113134)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 404.3138313831383)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 389.18862655496315)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 374.063421726788)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 358.9382168986129)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 343.81301207043776)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 328.68780724226264)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 313.5626024140875)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 298.4373975859124)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 283.31219275773725)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 268.1869879295621)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 253.061783101387)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 237.93657827321186)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 222.81137344503674)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 207.6861686168616)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 192.56096378868648)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 177.43575896051135)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 162.31055413233622)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 147.1853493041611)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 132.06014447598596)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 116.93493964781089)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 101.80973481963576)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 86.68452999146064)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 71.55932516328556)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(173.80952380952382, 56.434120335110435)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 548.0032772508021)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 487.5024579381015)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 536.6593736296707)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 476.15855431697014)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 523.4248194050174)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 462.9240000923169)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(128.78787878787878, 493.17440974866713)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(117.53246753246752, 463.86932539407786)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 396.7512289690507)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(106.27705627705627, 441.6541808026957)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 385.40732534791937)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 321.1252048281751)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 290.87479517182487)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 215.24877103094917)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 169.8731565464238)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(95.02164502164501, 413.5307530753075)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 305.99999999999994)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 260.62438551547456)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 203.90486740981783)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 158.52925292529244)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 94.2471324055482)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(83.76623376623377, 386.2344849869602)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 283.31219275773725)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 181.21706016755513)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 124.4975420618984)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 82.90322878441685)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(72.5108225108225, 365.023748528699)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(128.78787878787878, 260.62438551547456)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(128.78787878787878, 152.8573011147268)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 69.66867455976364)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(61.25541125541126, 312.82406702208675)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(117.53246753246752, 111.26298783724519)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(50.0, 212.04352742966597)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(50.0, 111.26298783724519)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(50.0, 312.82406702208675)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(117.53246753246752, 69.66867455976364)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(117.53246753246752, 152.8573011147268)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(61.25541125541126, 260.62438551547456)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(61.25541125541126, 365.023748528699)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 56.434120335110435)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 82.90322878441685)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(128.78787878787878, 124.4975420618984)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(128.78787878787878, 181.21706016755513)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(128.78787878787878, 237.93657827321186)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(128.78787878787878, 283.31219275773725)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(72.5108225108225, 343.81301207043776)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(72.5108225108225, 386.2344849869602)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 71.55932516328556)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 94.2471324055482)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 116.93493964781089)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 132.06014447598596)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 158.52925292529244)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 203.90486740981783)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 260.62438551547456)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 305.99999999999994)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(83.76623376623377, 358.9382168986129)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(83.76623376623377, 413.5307530753075)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 86.68452999146064)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 101.80973481963576)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 147.1853493041611)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 169.8731565464238)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 192.56096378868648)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 215.24877103094917)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 253.061783101387)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 268.1869879295621)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 290.87479517182487)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 321.1252048281751)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(95.02164502164501, 385.40732534791937)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(95.02164502164501, 441.6541808026957)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 162.31055413233622)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 177.43575896051135)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 207.6861686168616)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 222.81137344503674)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 283.31219275773725)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 298.4373975859124)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 313.5626024140875)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 328.68780724226264)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 374.063421726788)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 396.7512289690507)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(106.27705627705627, 419.4390362113134)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(106.27705627705627, 463.86932539407786)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 389.18862655496315)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 404.3138313831383)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(117.53246753246752, 434.56424103948854)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(117.53246753246752, 493.17440974866713)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(128.78787878787878, 462.9240000923169)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(128.78787878787878, 523.4248194050174)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 449.68944586766366)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 476.15855431697014)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 510.1902651803642)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(140.04329004329003, 536.6593736296707)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 464.81465069583885)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 487.5024579381015)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 525.3154700085394)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(151.2987012987013, 548.0032772508021)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 479.9398555240139)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 495.06506035218905)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 540.4406748367144)\"><circle r=\"0.0\"></circle></g><g class=\"toyplot-Datum\" style=\"fill:rgb(40%,76.1%,64.7%);fill-opacity:1.0;opacity:1.0;stroke:rgb(40%,76.1%,64.7%);stroke-opacity:1.0\" transform=\"translate(162.55411255411255, 555.5658796648896)\"><circle r=\"0.0\"></circle></g></g></g><g class=\"toyplot-mark-Text\" id=\"tcd7a1e7394204eaaa71efbc3cc8e7e2b\"><g class=\"toyplot-Series\"><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,555.5658796648896)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">deflexus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,540.4406748367144)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">viridis_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,525.3154700085394)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">muricatus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,510.1902651803642)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">fimbriatus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,495.06506035218905)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">asplundii_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,479.9398555240139)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">graecizans_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,464.81465069583885)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">tricolor_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,449.68944586766366)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">standleyanus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,434.56424103948854)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">blitum_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,419.4390362113134)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">palmeri_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,404.3138313831383)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">californicus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,389.18862655496315)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">albus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,374.063421726788)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">australis_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,358.9382168986129)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">cannabinus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,343.81301207043776)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">crassipes_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,328.68780724226264)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">arenicola_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,313.5626024140875)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">tuberculatus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,298.4373975859124)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">floridanus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,283.31219275773725)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">watsonii_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,268.1869879295621)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">greggii_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,253.061783101387)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">acanthochiton_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,237.93657827321186)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">blitoides_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,222.81137344503674)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">cruentus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,207.6861686168616)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">hybridus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,192.56096378868648)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">hypochondriacus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,177.43575896051135)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">dubius_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,162.31055413233622)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">spinosus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,147.1853493041611)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">reference</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,132.06014447598596)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">quitensis_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,116.93493964781089)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">caudatus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,101.80973481963576)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">wrightii_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,86.68452999146064)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">retroflexus_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,71.55932516328556)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">powellii_pooled</text></g><g class=\"toyplot-Datum\" transform=\"translate(173.80952380952382,56.434120335110435)\"><text style=\"fill:rgb(14.9%,14.9%,14.9%);fill-opacity:1.0;font-family:helvetica;font-size:11.0px;font-weight:normal;opacity:1;stroke:none;vertical-align:baseline;white-space:pre\" x=\"15.0\" y=\"2.8104999999999993\">acutilobus_pooled</text></g></g></g></g></g></svg><div class=\"toyplot-behavior\"><script>(function()\n",
       "{\n",
       "var modules={};\n",
       "modules[\"toyplot/tables\"] = (function()\n",
       "    {\n",
       "        var tables = [];\n",
       "\n",
       "        var module = {};\n",
       "\n",
       "        module.set = function(owner, key, names, columns)\n",
       "        {\n",
       "            tables.push({owner: owner, key: key, names: names, columns: columns});\n",
       "        }\n",
       "\n",
       "        module.get = function(owner, key)\n",
       "        {\n",
       "            for(var i = 0; i != tables.length; ++i)\n",
       "            {\n",
       "                var table = tables[i];\n",
       "                if(table.owner != owner)\n",
       "                    continue;\n",
       "                if(table.key != key)\n",
       "                    continue;\n",
       "                return {names: table.names, columns: table.columns};\n",
       "            }\n",
       "        }\n",
       "\n",
       "        module.get_csv = function(owner, key)\n",
       "        {\n",
       "            var table = module.get(owner, key);\n",
       "            if(table != undefined)\n",
       "            {\n",
       "                var csv = \"\";\n",
       "                csv += table.names.join(\",\") + \"\\n\";\n",
       "                for(var i = 0; i != table.columns[0].length; ++i)\n",
       "                {\n",
       "                  for(var j = 0; j != table.columns.length; ++j)\n",
       "                  {\n",
       "                    if(j)\n",
       "                      csv += \",\";\n",
       "                    csv += table.columns[j][i];\n",
       "                  }\n",
       "                  csv += \"\\n\";\n",
       "                }\n",
       "                return csv;\n",
       "            }\n",
       "        }\n",
       "\n",
       "        return module;\n",
       "    })();\n",
       "modules[\"toyplot/root/id\"] = \"t165ed59554964f419b1ef2bb70fc18ce\";\n",
       "modules[\"toyplot/root\"] = (function(root_id)\n",
       "    {\n",
       "        return document.querySelector(\"#\" + root_id);\n",
       "    })(modules[\"toyplot/root/id\"]);\n",
       "modules[\"toyplot/canvas/id\"] = \"tb36a481816bc4a8fa378ad649a925047\";\n",
       "modules[\"toyplot/canvas\"] = (function(canvas_id)\n",
       "    {\n",
       "        return document.querySelector(\"#\" + canvas_id);\n",
       "    })(modules[\"toyplot/canvas/id\"]);\n",
       "modules[\"toyplot/menus/context\"] = (function(root, canvas)\n",
       "    {\n",
       "        var wrapper = document.createElement(\"div\");\n",
       "        wrapper.innerHTML = \"<ul class='toyplot-context-menu' style='background:#eee; border:1px solid #b8b8b8; border-radius:5px; box-shadow: 0px 0px 8px rgba(0%,0%,0%,0.25); margin:0; padding:3px 0; position:fixed; visibility:hidden;'></ul>\"\n",
       "        var menu = wrapper.firstChild;\n",
       "\n",
       "        root.appendChild(menu);\n",
       "\n",
       "        var items = [];\n",
       "\n",
       "        var ignore_mouseup = null;\n",
       "        function open_menu(e)\n",
       "        {\n",
       "            var show_menu = false;\n",
       "            for(var index=0; index != items.length; ++index)\n",
       "            {\n",
       "                var item = items[index];\n",
       "                if(item.show(e))\n",
       "                {\n",
       "                    item.item.style.display = \"block\";\n",
       "                    show_menu = true;\n",
       "                }\n",
       "                else\n",
       "                {\n",
       "                    item.item.style.display = \"none\";\n",
       "                }\n",
       "            }\n",
       "\n",
       "            if(show_menu)\n",
       "            {\n",
       "                ignore_mouseup = true;\n",
       "                menu.style.left = (e.clientX + 1) + \"px\";\n",
       "                menu.style.top = (e.clientY - 5) + \"px\";\n",
       "                menu.style.visibility = \"visible\";\n",
       "                e.stopPropagation();\n",
       "                e.preventDefault();\n",
       "            }\n",
       "        }\n",
       "\n",
       "        function close_menu()\n",
       "        {\n",
       "            menu.style.visibility = \"hidden\";\n",
       "        }\n",
       "\n",
       "        function contextmenu(e)\n",
       "        {\n",
       "            open_menu(e);\n",
       "        }\n",
       "\n",
       "        function mousemove(e)\n",
       "        {\n",
       "            ignore_mouseup = false;\n",
       "        }\n",
       "\n",
       "        function mouseup(e)\n",
       "        {\n",
       "            if(ignore_mouseup)\n",
       "            {\n",
       "                ignore_mouseup = false;\n",
       "                return;\n",
       "            }\n",
       "            close_menu();\n",
       "        }\n",
       "\n",
       "        function keydown(e)\n",
       "        {\n",
       "            if(e.key == \"Escape\" || e.key == \"Esc\" || e.keyCode == 27)\n",
       "            {\n",
       "                close_menu();\n",
       "            }\n",
       "        }\n",
       "\n",
       "        canvas.addEventListener(\"contextmenu\", contextmenu);\n",
       "        canvas.addEventListener(\"mousemove\", mousemove);\n",
       "        document.addEventListener(\"mouseup\", mouseup);\n",
       "        document.addEventListener(\"keydown\", keydown);\n",
       "\n",
       "        var module = {};\n",
       "        module.add_item = function(label, show, activate)\n",
       "        {\n",
       "            var wrapper = document.createElement(\"div\");\n",
       "            wrapper.innerHTML = \"<li class='toyplot-context-menu-item' style='background:#eee; color:#333; padding:2px 20px; list-style:none; margin:0; text-align:left;'>\" + label + \"</li>\"\n",
       "            var item = wrapper.firstChild;\n",
       "\n",
       "            items.push({item: item, show: show});\n",
       "\n",
       "            function mouseover()\n",
       "            {\n",
       "                this.style.background = \"steelblue\";\n",
       "                this.style.color = \"white\";\n",
       "            }\n",
       "\n",
       "            function mouseout()\n",
       "            {\n",
       "                this.style.background = \"#eee\";\n",
       "                this.style.color = \"#333\";\n",
       "            }\n",
       "\n",
       "            function choose_item(e)\n",
       "            {\n",
       "                close_menu();\n",
       "                activate();\n",
       "\n",
       "                e.stopPropagation();\n",
       "                e.preventDefault();\n",
       "            }\n",
       "\n",
       "            item.addEventListener(\"mouseover\", mouseover);\n",
       "            item.addEventListener(\"mouseout\", mouseout);\n",
       "            item.addEventListener(\"mouseup\", choose_item);\n",
       "            item.addEventListener(\"contextmenu\", choose_item);\n",
       "\n",
       "            menu.appendChild(item);\n",
       "        };\n",
       "        return module;\n",
       "    })(modules[\"toyplot/root\"],modules[\"toyplot/canvas\"]);\n",
       "modules[\"toyplot/io\"] = (function()\n",
       "    {\n",
       "        var module = {};\n",
       "        module.save_file = function(mime_type, charset, data, filename)\n",
       "        {\n",
       "            var uri = \"data:\" + mime_type + \";charset=\" + charset + \",\" + data;\n",
       "            uri = encodeURI(uri);\n",
       "\n",
       "            var link = document.createElement(\"a\");\n",
       "            if(typeof link.download != \"undefined\")\n",
       "            {\n",
       "              link.href = uri;\n",
       "              link.style = \"visibility:hidden\";\n",
       "              link.download = filename;\n",
       "\n",
       "              document.body.appendChild(link);\n",
       "              link.click();\n",
       "              document.body.removeChild(link);\n",
       "            }\n",
       "            else\n",
       "            {\n",
       "              window.open(uri);\n",
       "            }\n",
       "        };\n",
       "        return module;\n",
       "    })();\n",
       "(function(tables, context_menu, io, owner_id, key, label, names, columns, filename)\n",
       "        {\n",
       "            tables.set(owner_id, key, names, columns);\n",
       "\n",
       "            var owner = document.querySelector(\"#\" + owner_id);\n",
       "            function show_item(e)\n",
       "            {\n",
       "                return owner.contains(e.target);\n",
       "            }\n",
       "\n",
       "            function choose_item()\n",
       "            {\n",
       "                io.save_file(\"text/csv\", \"utf-8\", tables.get_csv(owner_id, key), filename + \".csv\");\n",
       "            }\n",
       "\n",
       "            context_menu.add_item(\"Save \" + label + \" as CSV\", show_item, choose_item);\n",
       "        })(modules[\"toyplot/tables\"],modules[\"toyplot/menus/context\"],modules[\"toyplot/io\"],\"tb9a5a6944b6047f18faa6a077899d89a\",\"vertex_data\",\"graph vertex data\",[\"x\", \"y\"],[[-0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -1.0, -1.0, -2.0, -2.0, -3.0, -3.0, -4.0, -5.0, -1.0, -6.0, -2.0, -1.0, -1.0, -1.0, -1.0, -7.0, -2.0, -1.0, -2.0, -2.0, -1.0, -8.0, -3.0, -3.0, -1.0, -2.0, -9.0, -4.0, -4.0, -3.0, -10.0, -5.0, -11.0, -11.0, -11.0, -5.0, -5.0, -10.0, -10.0, -3.0, -3.0, -4.0, -4.0, -4.0, -4.0, -9.0, -9.0, -2.0, -2.0, -1.0, -1.0, -3.0, -3.0, -3.0, -3.0, -8.0, -8.0, -1.0, -1.0, -2.0, -2.0, -2.0, -2.0, -1.0, -1.0, -2.0, -2.0, -7.0, -7.0, -1.0, -1.0, -1.0, -1.0, -1.0, -1.0, -1.0, -1.0, -2.0, -2.0, -6.0, -6.0, -1.0, -1.0, -5.0, -5.0, -4.0, -4.0, -3.0, -3.0, -3.0, -3.0, -2.0, -2.0, -2.0, -2.0, -1.0, -1.0, -1.0, -1.0], [0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0, 17.0, 18.0, 19.0, 20.0, 21.0, 22.0, 23.0, 24.0, 25.0, 26.0, 27.0, 28.0, 29.0, 30.0, 31.0, 32.0, 33.0, 0.5, 4.5, 1.25, 5.25, 2.125, 6.125, 4.125, 6.0625, 10.5, 7.53125, 11.25, 15.5, 17.5, 22.5, 25.5, 9.390625, 16.5, 19.5, 23.25, 26.25, 30.5, 11.1953125, 18.0, 24.75, 28.5, 31.25, 12.59765625, 19.5, 26.625, 32.125, 16.048828125, 29.375, 22.7119140625, 29.375, 16.048828125, 32.125, 26.625, 19.5, 12.59765625, 33.0, 31.25, 28.5, 24.75, 21.0, 18.0, 14.0, 11.1953125, 32.0, 30.5, 29.0, 28.0, 26.25, 23.25, 19.5, 16.5, 13.0, 9.390625, 31.0, 30.0, 27.0, 25.5, 24.0, 22.5, 20.0, 19.0, 17.5, 15.5, 11.25, 7.53125, 26.0, 25.0, 23.0, 22.0, 18.0, 17.0, 16.0, 15.0, 12.0, 10.5, 9.0, 6.0625, 11.0, 10.0, 8.0, 4.125, 6.125, 2.125, 7.0, 5.25, 3.0, 1.25, 6.0, 4.5, 2.0, 0.5, 5.0, 4.0, 1.0, 0.0]],\"toyplot\");\n",
       "(function(tables, context_menu, io, owner_id, key, label, names, columns, filename)\n",
       "        {\n",
       "            tables.set(owner_id, key, names, columns);\n",
       "\n",
       "            var owner = document.querySelector(\"#\" + owner_id);\n",
       "            function show_item(e)\n",
       "            {\n",
       "                return owner.contains(e.target);\n",
       "            }\n",
       "\n",
       "            function choose_item()\n",
       "            {\n",
       "                io.save_file(\"text/csv\", \"utf-8\", tables.get_csv(owner_id, key), filename + \".csv\");\n",
       "            }\n",
       "\n",
       "            context_menu.add_item(\"Save \" + label + \" as CSV\", show_item, choose_item);\n",
       "        })(modules[\"toyplot/tables\"],modules[\"toyplot/menus/context\"],modules[\"toyplot/io\"],\"tb9a5a6944b6047f18faa6a077899d89a\",\"edge_data\",\"graph edge data\",[\"source\", \"target\"],[[67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 66, 66, 65, 65, 64, 64, 63, 63, 62, 62, 61, 61, 60, 60, 59, 59, 58, 58, 57, 57, 56, 56, 55, 55, 54, 54, 53, 53, 52, 52, 51, 51, 50, 50, 49, 49, 48, 48, 47, 47, 46, 46, 45, 45, 44, 44, 43, 43, 42, 42, 41, 41, 40, 40, 39, 39, 38, 38, 37, 37, 36, 36, 35, 35, 34, 34], [65, 64, 63, 62, 61, 60, 33, 59, 58, 57, 21, 56, 14, 55, 32, 54, 29, 28, 53, 52, 51, 50, 13, 49, 31, 30, 27, 48, 24, 47, 20, 19, 46, 45, 44, 43, 26, 25, 23, 22, 18, 17, 16, 15, 12, 42, 9, 41, 11, 10, 8, 40, 39, 38, 7, 37, 3, 36, 6, 35, 2, 34, 5, 4, 1, 0, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132]],\"toyplot\");\n",
       "})();</script></div></div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import toytree\n",
    "\n",
    "tre = toytree.tree(\"/home/sandra/Files/MrBait/RAxML_bipartitions.cp_RAX\")\n",
    "tre.root([\"quitensis_pooled\",\"caudatus_pooled\",\"dubius_pooled\",\"spinosus_pooled\",\"reference\",\"cruentus_pooled\",\n",
    "          \"hybridus_pooled\",\"hypochondriacus_pooled\",\"wrightii_pooled\",\"retroflexus_pooled\",\"powellii_pooled\",\"acutilobus_pooled\"]).draw(use_edge_lengths=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Start with MrBait"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "\t=======================================================================\r\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\r\n",
      "\t=======================================================================\r\n",
      "\r\n",
      "\tVersion:  1.2.6\r\n",
      "\tAuthor:   Tyler K. Chafin\r\n",
      "\tContact:  tkchafin@uark.edu\r\n",
      "\tLicense:  GNU Public License v3.0\r\n",
      "\r\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\r\n",
      "\r\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\r\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\r\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\r\n",
      "\r\n",
      "\tSource code: https://github.com/tkchafin/mrbait\r\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\r\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\r\n",
      "\r\n",
      "\t=======================================================================\r\n",
      "\t\r\n",
      "\r\n",
      "\tParsing command line arguments...\r\n",
      "\tExiting because help menu was called.\r\n",
      "\r\n",
      "MrBait Help Menu\r\n",
      "\r\n",
      "For more detailed information, please see the manual.\r\n",
      "\r\n",
      "\r\n",
      "Usage:\r\n",
      "\r\n",
      "\t /moto/home/slh2181/miniconda3/bin/mrbait -A </path/to/MAF> <-c><-l><-t><...>\r\n",
      "\r\n",
      "Description:\r\n",
      "\r\n",
      "\tMrBait is a program for designing baits for targeted enrichment studies\r\n",
      " \tincluding Ultraconverved Elements (UCEs), Anchored enrichment, and RAD-capture.\r\n",
      " \tIt can also be used to customize bait design for any of these methods based on\r\n",
      " \ta variety of built-in methods and constraints.\r\n",
      "\r\n",
      "\tNOTE: SQLite databases are currently remade from scratch with each\r\n",
      " \tinstance of the program. Support for opening and editing older database files\r\n",
      " \twill come soon.\r\n",
      "\r\n",
      "\r\n",
      "General options:\r\n",
      "\r\n",
      "\t-r,--resume\t: Resumes from a database pre-loaded with alignments (integer)\r\n",
      "\t\t--Options\r\n",
      "\t\t\t-r 1 : Continues pipeline beginning after Step 1 (Loading alignments)\r\n",
      "\t\t\t-r 2 : Continues after Step 2 (Target discovery)\r\n",
      "\t\t\t-r 3 : Continues after Step 3 (Target filtering/ selection)\r\n",
      "\t\t\t-r 4 : Continues after step 4 (Bait discovery)\r\n",
      "\t--db\t\t: .sqlite file containing pre-existing database. For use with --resume\r\n",
      "\t-T,--threads\t: Number of threads to use for processes that can run in parallel [1]\r\n",
      "\t-h,--help\t: Displays this help menu\r\n",
      "\t\r\n",
      "\r\n",
      "Input options:\r\n",
      "\r\n",
      "\t-M,--maf\t: Input multiple alignment MAF file\r\n",
      "\t-L,--loci\t: For RAD-data, as the \".loci\" output of pyRAD\r\n",
      "\t-A,--assembly\t: Input whole genome assembly as FASTA\r\n",
      "\t-X,--xmfa\t: Input whole genome alignments as XMFA\r\n",
      "\r\n",
      "Assembly input options (for use only with -A <genome.fasta>):\r\n",
      "\r\n",
      "\t-V,--vcf\t: VCF file containing variant information\r\n",
      "\t-G,--gff\t: GFF file containing annotation information\r\n",
      "\t-B,--bed\t: BED file containing regions (only 1 header line)\r\n",
      "\t--vcfALT\t: If using VCF and reference allele is gap/N, attempt\r\n",
      "\t\t \t   to call new consensus using VCF ALT alleles [default=OFF]\r\n",
      "\r\n",
      "Alignment filtering/ consensus options (use with -M or -L inputs):\r\n",
      "\r\n",
      "\t-c,--cov\t: Minimum number of sequences per alignment, for MAF or LOCI input [1]\r\n",
      "\t-l,--len\t: Minimum alignment length to attempt bait design [80]\r\n",
      "\t-q,--thresh\t: Threshold proportion for gap/N to include in consensus [0.1]\r\n",
      "\t-Q,--max_ambig\t: Maximum proportion of gap/N bases allowed in a consensus sequence [0.5]\r\n",
      "\t-k,--mask\t: Threshold proportion for masking (lower case) to include in consensus [0.1]\r\n",
      "\t-K,--max_mask\t: Maximum proportion of masked bases allowed in a consensus sequence [0.5]\r\n",
      "\t--consens_maf\t: Threshold minor allele frequency to code a SNP in consensus sequence [0.0]\r\n",
      "\t--dustMask\t: Use VSEARCH to mask consensus sequences using the DUST algorithm\r\n",
      "\r\n",
      "General Bait Design options:\r\n",
      "\r\n",
      "\t-b, --bait\t: Bait length [default=80]\r\n",
      "\t-w, --win_shift\t: Sliding window shift distance [1]\r\n",
      "\t\t  --Increasing will speed up program but could lower accuracy\r\n",
      "\t\t  --Cannot be longer than bait length <-b,--bait>\r\n",
      "\t-R,--mult_reg\t: Allow multiple target regions per locus [false]\r\n",
      "\t\t--See \"Target Region options\" below\r\n",
      "\t-m,--min_mult\t: Minimum alignment length to allow multiple regions\r\n",
      "\t\t--By default is set to the value of <-l>. Only applies when <-R>\r\n",
      "\t\t--When -R and -T are off, only one bait is selected per locus\r\n",
      "\t-v,--var_max\t: Maximum number of SNP columns to be allowed in a bait [0]\r\n",
      "\t\t--These will be coded as the appropriate IUPAC ambiguity code\r\n",
      "\t\t--Can be expanded in final output using the <--expand> flag\r\n",
      "\t-n,--numN\t: Number of consensus Ns allowed in a bait [0]\r\n",
      "\t\t--These will be inserted as \"N\" unless <-N> is used\r\n",
      "\t-g,--numG\t: Number of consensus indels (\"-\") allowed in bait [0]\r\n",
      "\r\n",
      "Target Region options:\r\n",
      "\r\n",
      "\t-D,--dist_r\t: Minimum distance between target regions [100]\r\n",
      "\t\t--Conflicts will be resolved according to --select_r\r\n",
      "\t-d,--flank_dist\t: Distance in flanking regions selection and filtering targets\r\n",
      "\t\t--See relevant options in <--select_r> and <--filter_r>\r\n",
      "\t\t--By default will be set to [500]\r\n",
      "\t\t--Appopriate value likely will correspond to your sequencing insert size\r\n",
      "\t--target_all\t: Skip target region search, and use full (passing) loci for bait design\r\n",
      "\t\t--Note that --select_r and --filter_r methods will still be applied\r\n",
      "\t-S, --select_r\t: Which criterion to select target regions w/in <-D>\r\n",
      "\t\t--Options\r\n",
      "\t\t\tsnp          : Most SNPs w/in \"d\" bases\r\n",
      "\t\t\tbad          : Least Ns and gaps w/n \"d\" bases\r\n",
      "\t\t\tcons         : Most conserved w/in \"d\" bases\r\n",
      "\t\t\trand         : Randomly choose a bait region [default]\r\n",
      "\t\t\tEx: -S snp -d 100 to choose region with most SNPs w/in 100 bases\r\n",
      "\t-F,--filter_r\t: Include any criteria used to filter ALL bait regions\r\n",
      "\t\t--Options\r\n",
      "\t\t\tlen=[x,y]    : Target length between \"x\" (min) and \"y\" (max)\r\n",
      "\t\t\tgap=[x]      : Maximum of \"x\" indels in target region\r\n",
      "\t\t\tbad=[x]      : Maximum of \"x\" Ns in target region\r\n",
      "\t\t\tsnp=[x,y]    : Min. of \"x\" and max \"y\" SNPs within --flank_dist\r\n",
      "\t\t\tmask=[x]     : Maximum of \"x\" proportion of masked bases allowed\r\n",
      "\t\t\tgc=[x,y]     : Proportion of G/C bases between \"x\" (min) and \"y\" (max)\r\n",
      "\t\t\trand=[x]     : Randomly retain \"x\" target regions w/ baits\r\n",
      "\t\t\tpw=[i,q]     : Pairwise alignment, removing when \"i\" identity in \"q\" proportion\r\n",
      "\t\t\tblast_i=[i,q]: Only retain hits over \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\tblast_x=[i,q]: Remove hits over \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\tblast_a=[i,q]: Remove targets having >1 hits with \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\tgff=[type]   : Only retain targets within \"d\" distance of \"type\" GFF elements\r\n",
      "\t\t\t               Use \"all\" to target all types, or see GFF3 specifications for SO feature types:\r\n",
      "\t\t\t               https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md\r\n",
      "\t\t\tgff_a=[alias]: Only retain targets within \"d\" distance of GFF records with attributes of \"Alias\"\r\n",
      "\t\t\t               Note that \"gff\" and \"gffa\" options are case-insensitive\r\n",
      "\t\t\tbed_x        : Exclude targets within \"d\" distance of BED records\r\n",
      "\t\t\tbed_i        : Only include targets within \"d\" distance of BED records\r\n",
      "\t\tEx1: -F snp=1,10 -d 100 to sample when 1-10 SNPs w/in 100 bases\r\n",
      "\t\tEx2: -F gc=0.2,0.8 -F rand=100 to randomly sample 100 targets with GC between 20-80%\r\n",
      "\t\tEx3: -F mask=0.0,0.1 to remove targets with >10% \\masked bases\r\n",
      "\t\tEx4: -F gff=exon -d 1000 to constrain targets to within 1000 bases of exons\r\n",
      "\t\tNOTE: Values for gc, mask, pw, and blast* methods are given as proportions (e.g. 90% = 0.9)\r\n",
      "\r\n",
      "Bait Design / Selection options:\r\n",
      "\r\n",
      "\t-s,--select_b\t: Which criterion to select a bait for target region\r\n",
      "\t\t--NOTE: This option is ignored when <-W>\r\n",
      "\t\t--Options\r\n",
      "\t\t\ttile=[x]     : Tile baits with \"x\" overlapping bases [default; x= bait_length/2]\r\n",
      "\t\t\tcenter=[n,x] : Design \"n\" centered baits with \"x\" overlap\r\n",
      "\t\t\tflank=[n,x]  : Design \"x\" terminal baits (each side) with \"x\" overlap\r\n",
      "\t\t\tcalc=[n,x]   : Design \"n\" baits per target, with \"x\" maximum overlap\r\n",
      "\t\t\tEx: -s tile=40 to tile baits with an overlap of 40 bases\r\n",
      "\t-f,--filter_b\t: Include any criteria used to filter ALL baits\r\n",
      "\t\t--Options\r\n",
      "\t\t\tmask=[x]     : Maximum of \"x\" proportion of masked bases allowed\r\n",
      "\t\t\tgc=[x,y]     : Proportion of G/C bases between \"x\" (min) and \"y\" (max)\r\n",
      "\t\t\tpw=[i,q]     : Pairwise alignment, removing when \"i\" identity in \"q\" proportion\r\n",
      "\t\t\trc=[i,q]     : Pairwise alignment against reverse-complements (see -r pw=[i,q])\r\n",
      "\t\t\tblast_i=[i,q]: Only retain hits over \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\tblast_x=[i,q]: Remove hits over \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\tblast_a=[i,q]: Remove targets having >1 hits with \"i\" identity and \"q\" query coverage to provided db\r\n",
      "\t\t\trand=[x]     : Randomly retain \"x\" baits\r\n",
      "\t\t\tbed_x        : Exclude baits within \"d\" distance of BED records\r\n",
      "\t\t\tbed_i        : Only include baits within \"d\" distance of BED records\r\n",
      "\r\n",
      "VSEARCH Parameters (use when --select_b or --select_r = \"pw\" or \"rc\", or --dustMask):\r\n",
      "\r\n",
      "\t--vsearch\t: Path to VSEARCH executable if other than provided\r\n",
      "\t\t--MrBait will try to detect OS and appropriate exectable to use\r\n",
      "\t--vthreads\t: Number of threads for VSEARCH\r\n",
      "\t\t--Will set to the value of <--threads> if not defined\r\n",
      "\t--vsearch_qmask\t: Option for the --qmask option in VSEARCH\r\n",
      "\t\t--Only relevant when --select_b or --select_r = \"pw\" or \"rc\"\r\n",
      "\t\t--Options\r\n",
      "\t\t\tdust       : Mask sequences using the DUST algorithm [default]\r\n",
      "\t\t\tsoft       : Mask sequences using lower-cased characters\r\n",
      "\t\t\tnone       : Skip masking for pairwise alignments\r\n",
      "\r\n",
      "Graph Parameters (relevant for conflict resolution when --select_b or --select_r = \"pw\" or \"rc\"):\r\n",
      "\r\n",
      "\t--noGraph\t: Skip graph-based conflict resolution and remove ALL conflicting pairs\r\n",
      "\t--noWeightGraph\t: Unweighted (random) conflict resolution\r\n",
      "\t\t--MrBait will by default weight conflict resolution to maximize flanking SNPs\r\n",
      "\t--weightByMin\t: Weight loci by minimum ambiguities (NOT by flanking SNPs)\r\n",
      "\t--weightMax\t: Maximum graph size to attempt weighted algorithm\r\n",
      "\t\t--Default is 50,000. May be necessary for very large datasets with a lot of redundancy\r\n",
      "\r\n",
      "BLAST Parameters (use when --select_b or --select_r = \"blast\"):\r\n",
      "\r\n",
      "\t--blastdb\t: Path and prefix to existing Blast database to use\r\n",
      "\t--fastadb\t: Path and prefix to FASTA formatted sequences to make blast db\r\n",
      "\t--e_value\t: Minimum e-value cutoff for reporting BLAST hits [0.000001]\r\n",
      "\t--gapopen\t: Gap opening penalty for blastn [5]\r\n",
      "\t--gapextend\t: Gap extension penatly for blastn [2]\r\n",
      "\t--word_size\t: Word size for blastn\r\n",
      "\t--nodust\t: Turn off low-complexity filter for blastn\r\n",
      "\t--megablast\t: Use megablast rather than blastn\r\n",
      "\t\t--Default word size will be set to 28 if not provided\r\n",
      "\t--blastn\t: Path to blastn binary if different than default\r\n",
      "\t--makedb\t: Path to makeblastdb binary if different than default\r\n",
      "\t\t--MrBait will try to detect OS and appropriate exectable to use\r\n",
      "\t--max_hits\t: This value is passed directly to blast as --max_target_seqs\r\n",
      "\t\t--By default this is set at 10,000. Should only be necessary to change\r\n",
      "\t\t  if computation time becomes excessive with highly redundant databases\r\n",
      "\r\n",
      "Output options:\r\n",
      "\r\n",
      "\t-x,--expand\t: In output bait table, expand all ambiguities\r\n",
      "\t\t--Gaps are expanded as [ACGT] and absent\r\n",
      "\t\t--\"N\"s are expanded as [ACGT]\r\n",
      "\t--strand\t: Output strand. Options: \"+\", \"-\", \"both\"\r\n",
      "\t\t--\"-\" print antiparallel (reverse complement)\r\n",
      "\t-t,--print_tr\t: Boolean. Print target regions to FASTA file\r\n",
      "\t--print_loc\t: Boolean. Print consensus locus catalog to FASTA file\r\n",
      "\t-o,--out\t: Prefix for output files\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "! mrbait -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## This is what I submited for the baits, March 23, 2020"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.2.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /moto/eaton/users/slh2181/MrBait/bait_no_cp_9_45.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:00 (0.771315 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum alignment depth (-c): 5\n",
      "\t\t\tMinimum alignment length (-l): 100\n",
      "\t\t\tThreshold to call \"N\" or gap consensus base (-q): 0.15\n",
      "\t\t\tThreshold to retain masked base (-k): 0.15\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.55\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.55\n",
      "\t\t\tMinor allele frequency threshold (--consens_maf): 0.2\n",
      "\t\tLoading LOCI file: /moto/eaton/users/slh2181/MrBait/pooled_9_45_outfiles/pooled_9_45.loci\n",
      "\t\t\tLoading alignments using 24 parallel processes.\n",
      "\t\t\tReading 36256 alignments... This may take a while.\n",
      "\t\tMasking consensus sequences using the DUST algorithm in VSEARCH... Done!\n",
      "\t\tFiltering loci... Done!\n",
      "\n",
      "\t\t### Results: 6871 loci passed filtering! ###\n",
      "\t\tPrinting locus catalog to file...\n",
      "\t\t### Runtime: 1:13:41 (4421.996915 seconds) ###\n",
      "\n",
      "\tStep 2: Target Discovery\n",
      "\t\tStep 2 parameters:\n",
      "\t\t\tSliding window width (-b, --bait): 90\n",
      "\t\t\tSliding window shift distance (-w, --win_shift): 1\n",
      "\t\t\tVariable columns allowed per target (-v, --var_max): 3\n",
      "\t\t\tAmbiguities (N's) allowed per target (-n, --numN): 1\n",
      "\t\t\tGap characters allowed per target (-g, --numG): 1\n",
      "\t\t\tFlanking distance to parse (-d,--flank_dist): 50\n",
      "\t\tStarting sliding window target discovery of 6871 loci...\n",
      "\t\t\tFinding targets using 24 parallel processes...\n",
      "\n",
      "\t\t### Results: 5458 potential targets identified! ###\n",
      "\t\t### Runtime: 0:00:31 (31.129374 seconds) ###\n",
      "\n",
      "\tStep 3: Target Filtering and Selection\n",
      "\t\tStep 3 parameters:\n",
      "\t\t\tMultiple targets allowed per locus (-R, --mult_reg): True\n",
      "\t\t\tMinimum locus length for multiple targets (-m, --min_mult): 200\n",
      "\t\t\tMinimum distance between targets within locus (-D, --dist_r): 50\n",
      "\t\t\tFlanking distance for target selection (-d,--flank_dist): 50\n",
      "\t\tApplying filters to 5458 targets.\n",
      "\t\t\tFiltering criterion: Between 1 and 15 flanking SNPs\n",
      "\t\t\tFiltering criterion: Between 0.25 and 0.75 GC content\n",
      "\t\t\tFiltering criterion: Target length between 125 and 9999\n",
      "\t\t\tFiltering criterion: BLAST inclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_no_cp_9_45\n",
      "\t\t\t  --Percent identity: 0.5\n",
      "\t\t\t  --Query coverage: 0.5\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 24\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\tResolving edges...\n",
      "\t\t\t  --Resolve by graph: True\n",
      "\t\t\t  --MIS method: Weighted\n",
      "\t\t\t  --Weights: Maximum variation\n",
      "\t\tResolving conflicting targets... Found 233 needing resolution.\n",
      "\t\t\tSelection criterion (-S, --select_r): Most conserved w/in 50 bases\n",
      "\t\t\tNote: Remaining conflicts will be randomly resolved.\n",
      "\t\tPrinting targets to file...\n",
      "\n",
      "\t\t### Results: 1993 targets passed filtering! ###\n",
      "\t\t### Runtime: 0:00:17 (17.309273 seconds) ###\n",
      "\n",
      "\tStep 4: Bait discovery\n",
      "\t\tStep 4 parameters:\n",
      "\t\t\tBait length (-b, --blen): 90\n",
      "\t\t\tDesigning 2 baits per target with 20 maximum overlap\n",
      "\n",
      "\t\t### Results: 2511 potential baits identified! ###\n",
      "\t\t### Runtime: 0:00:13 (13.326021 seconds) ###\n",
      "\n",
      "\tStep 5: Bait filtering\n",
      "\t\tFound 2511 potential baits!\n",
      "\t\tApplying filters to 2511 baits...\n",
      "\t\t\tFiltering criterion: Between 0.25 and 0.75 GC content\n",
      "\t\t\tFiltering criterion: BLAST exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_no_cp_9_45\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\t  --OVERRIDING BLAST DB: /moto/eaton/users/slh2181/reference/hypochondriacus_cpDNA.fasta\n",
      "\t\t\tFiltering criterion: BLAST ambiguous map exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_no_cp_9_45\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 24\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\t 2435 passed filtering.\n",
      "\t\tFormatting for printing...\n",
      "\t\tFinal baits output to: /moto/eaton/users/slh2181/MrBait/bait_no_cp_9_45.fasta\n",
      "\n",
      "\t\t### Results: 2435 baits output to file! ###\n",
      "\t\t### Runtime: 0:00:01 (1.874122 seconds) ###\n",
      "\n",
      "\t=======================================================================\n",
      "\t### Final output: /moto/eaton/users/slh2181/MrBait/bait_no_cp_9_45.fasta ###\n",
      "\t### Runtime: 1:14:46 (4486.407945 seconds) ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#All but cpDNA\n",
    "#--db bait_no_cp.sqlite --resume 4\n",
    "# varmax=3, len=135: 2500 targets\n",
    "\n",
    "! mrbait -T 24 --vthreads 24 -L /moto/eaton/users/slh2181/MrBait/pooled_9_45_outfiles/pooled_9_45.loci --cov 5 -l 100 --thresh 0.15 -Q 0.55 --mask 0.15 --max_mask 0.55 -b 90 --var_max 3 --numN 1 --numG 1 --mult_reg --min_mult 200 --dist_r 50 --flank_dist 50 --select_r cons --filter_r snp=1,15 -s calc=2,20 --filter_r pw=0.8,0.8 --filter_b pw=0.8,0.8 --filter_r gc=0.25,0.75 --filter_b gc=0.25,0.75 --print_loc --print_tr --out bait_no_cp_9_45 --consens_maf 0.2 --filter_r len=125,9999 --dustMask --blastx_fdb /moto/eaton/users/slh2181/reference/hypochondriacus_cpDNA.fasta --filter_b blast_x=0.8,0.8 --fastadb /moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.fa --filter_r blast_i=0.5,0.5 --filter_b blast_a=0.8,0.8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### This was the older thing using the assembly from the old version of ipyrad"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.2.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /moto/eaton/users/slh2181/MrBait/bait_no_cp.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:00 (0.654878 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum alignment depth (-c): 6\n",
      "\t\t\tMinimum alignment length (-l): 100\n",
      "\t\t\tThreshold to call \"N\" or gap consensus base (-q): 0.15\n",
      "\t\t\tThreshold to retain masked base (-k): 0.15\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.55\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.55\n",
      "\t\t\tMinor allele frequency threshold (--consens_maf): 0.2\n",
      "\t\tLoading LOCI file: /moto/eaton/users/slh2181/MrBait/pooled-spp33_Hardmask_format.loci\n",
      "\t\t\tLoading alignments using 24 parallel processes.\n",
      "\t\t\tReading 49844 alignments... This may take a while.\n",
      "\t\tMasking consensus sequences using the DUST algorithm in VSEARCH... Done!\n",
      "\t\tFiltering loci... Done!\n",
      "\n",
      "\t\t### Results: 10153 loci passed filtering! ###\n",
      "\t\tPrinting locus catalog to file...\n",
      "\t\t### Runtime: 0:05:28 (328.587148 seconds) ###\n",
      "\n",
      "\tStep 2: Target Discovery\n",
      "\t\tStep 2 parameters:\n",
      "\t\t\tSliding window width (-b, --bait): 90\n",
      "\t\t\tSliding window shift distance (-w, --win_shift): 1\n",
      "\t\t\tVariable columns allowed per target (-v, --var_max): 2\n",
      "\t\t\tAmbiguities (N's) allowed per target (-n, --numN): 0\n",
      "\t\t\tGap characters allowed per target (-g, --numG): 1\n",
      "\t\t\tFlanking distance to parse (-d,--flank_dist): 50\n",
      "\t\tStarting sliding window target discovery of 10153 loci...\n",
      "\t\t\tFinding targets using 24 parallel processes...\n",
      "\n",
      "\t\t### Results: 6632 potential targets identified! ###\n",
      "\t\t### Runtime: 0:00:42 (42.450465 seconds) ###\n",
      "\n",
      "\tStep 3: Target Filtering and Selection\n",
      "\t\tStep 3 parameters:\n",
      "\t\t\tMultiple targets allowed per locus (-R, --mult_reg): True\n",
      "\t\t\tMinimum locus length for multiple targets (-m, --min_mult): 200\n",
      "\t\t\tMinimum distance between targets within locus (-D, --dist_r): 50\n",
      "\t\t\tFlanking distance for target selection (-d,--flank_dist): 50\n",
      "\t\tApplying filters to 6632 targets.\n",
      "\t\t\tFiltering criterion: Between 1 and 15 flanking SNPs\n",
      "\t\t\tFiltering criterion: Between 0.27 and 0.73 GC content\n",
      "\t\t\tFiltering criterion: Target length between 125 and 9999\n",
      "\t\t\tFiltering criterion: BLAST inclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_no_cp\n",
      "\t\t\t  --Percent identity: 0.5\n",
      "\t\t\t  --Query coverage: 0.5\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 24\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\tResolving edges...\n",
      "\t\t\t  --Resolve by graph: True\n",
      "\t\t\t  --MIS method: Weighted\n",
      "\t\t\t  --Weights: Maximum variation\n",
      "\t\tResolving conflicting targets... Found 160 needing resolution.\n",
      "\t\t\tSelection criterion (-S, --select_r): Most conserved w/in 50 bases\n",
      "\t\t\tNote: Remaining conflicts will be randomly resolved.\n",
      "\t\tPrinting targets to file...\n",
      "\n",
      "\t\t### Results: 2079 targets passed filtering! ###\n",
      "\t\t### Runtime: 0:00:17 (17.783663 seconds) ###\n",
      "\n",
      "\tStep 4: Bait discovery\n",
      "\t\tStep 4 parameters:\n",
      "\t\t\tBait length (-b, --blen): 90\n",
      "\t\t\tDesigning 2 baits per target with 20 maximum overlap\n",
      "\n",
      "\t\t### Results: 2553 potential baits identified! ###\n",
      "\t\t### Runtime: 0:00:14 (14.533554 seconds) ###\n",
      "\n",
      "\tStep 5: Bait filtering\n",
      "\t\tFound 2553 potential baits!\n",
      "\t\tApplying filters to 2553 baits...\n",
      "\t\t\tFiltering criterion: Between 0.27 and 0.73 GC content\n",
      "\t\t\tFiltering criterion: BLAST exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_no_cp\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\t  --OVERRIDING BLAST DB: /moto/eaton/users/slh2181/reference/hypochondriacus_cpDNA.fasta\n",
      "\t\t\tFiltering criterion: BLAST ambiguous map exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_no_cp\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 24\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\t 2438 passed filtering.\n",
      "\t\tFormatting for printing...\n",
      "\t\tFinal baits output to: /moto/eaton/users/slh2181/MrBait/bait_no_cp.fasta\n",
      "\n",
      "\t\t### Results: 2438 baits output to file! ###\n",
      "\t\t### Runtime: 0:00:01 (1.884112 seconds) ###\n",
      "\n",
      "\t=======================================================================\n",
      "\t### Final output: /moto/eaton/users/slh2181/MrBait/bait_no_cp.fasta ###\n",
      "\t### Runtime: 0:06:45 (405.894732 seconds) ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#All but cpDNA...THIS\n",
    "#--db bait_no_cp.sqlite --resume 4\n",
    "# varmax=2, len=125: 2000 targets\n",
    "\n",
    "! mrbait -T 24 --vthreads 24 -L /moto/eaton/users/slh2181/MrBait/pooled-spp33_Hardmask_format.loci --cov 6 -l 100 --thresh 0.15 -Q 0.55 --mask 0.15 --max_mask 0.55 -b 90 --var_max 2 --numN 0 --numG 1 --mult_reg --min_mult 200 --dist_r 50 --flank_dist 50 --select_r cons --filter_r snp=1,15 -s calc=2,20 --filter_r pw=0.8,0.8 --filter_b pw=0.8,0.8 --filter_r gc=0.27,0.73 --filter_b gc=0.27,0.73 --print_loc --print_tr --out bait_no_cp --consens_maf 0.2 --filter_r len=125,9999 --dustMask --blastx_fdb /moto/eaton/users/slh2181/reference/hypochondriacus_cpDNA.fasta --filter_b blast_x=0.8,0.8 --fastadb /moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.fa --filter_r blast_i=0.5,0.5 --filter_b blast_a=0.8,0.8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## This is what I submited for chloroplast baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.2.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /moto/eaton/users/slh2181/MrBait/bait_cp_only_9_45.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:00 (0.022743 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum alignment depth (-c): 2\n",
      "\t\t\tMinimum alignment length (-l): 100\n",
      "\t\t\tThreshold to call \"N\" or gap consensus base (-q): 0.15\n",
      "\t\t\tThreshold to retain masked base (-k): 0.15\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.55\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.55\n",
      "\t\t\tMinor allele frequency threshold (--consens_maf): 0.2\n",
      "\t\tLoading LOCI file: /moto/eaton/users/slh2181/MrBait/pooled_9_45_outfiles/pooled_9_45.loci\n",
      "\t\t\tLoading alignments using 24 parallel processes.\n",
      "\t\t\tReading 36256 alignments... This may take a while.\n",
      "\t\tMasking consensus sequences using the DUST algorithm in VSEARCH... Done!\n",
      "\t\tFiltering loci... Done!\n",
      "\n",
      "\t\t### Results: 8652 loci passed filtering! ###\n",
      "\t\tPrinting locus catalog to file...\n",
      "\t\t### Runtime: 1:15:48 (4548.634963 seconds) ###\n",
      "\n",
      "\tStep 2: Target Discovery\n",
      "\t\tStep 2 parameters:\n",
      "\t\t\tSliding window width (-b, --bait): 90\n",
      "\t\t\tSliding window shift distance (-w, --win_shift): 1\n",
      "\t\t\tVariable columns allowed per target (-v, --var_max): 3\n",
      "\t\t\tAmbiguities (N's) allowed per target (-n, --numN): 1\n",
      "\t\t\tGap characters allowed per target (-g, --numG): 1\n",
      "\t\t\tFlanking distance to parse (-d,--flank_dist): 50\n",
      "\t\tStarting sliding window target discovery of 8652 loci...\n",
      "\t\t\tFinding targets using 24 parallel processes...\n",
      "\n",
      "\t\t### Results: 6776 potential targets identified! ###\n",
      "\t\t### Runtime: 0:00:31 (31.573902 seconds) ###\n",
      "\n",
      "\tStep 3: Target Filtering and Selection\n",
      "\t\tStep 3 parameters:\n",
      "\t\t\tMultiple targets allowed per locus (-R, --mult_reg): True\n",
      "\t\t\tMinimum locus length for multiple targets (-m, --min_mult): 200\n",
      "\t\t\tMinimum distance between targets within locus (-D, --dist_r): 50\n",
      "\t\t\tFlanking distance for target selection (-d,--flank_dist): 50\n",
      "\t\tApplying filters to 6776 targets.\n",
      "\t\t\tFiltering criterion: Between 1 and 20 flanking SNPs\n",
      "\t\t\tFiltering criterion: Between 0.25 and 0.75 GC content\n",
      "\t\t\tFiltering criterion: Target length between 100 and 9999\n",
      "\t\t\tFiltering criterion: BLAST inclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_cp_only_9_45\n",
      "\t\t\t  --Percent identity: 0.4\n",
      "\t\t\t  --Query coverage: 0.4\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 24\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\tResolving conflicting targets... Found 2 needing resolution.\n",
      "\t\t\tSelection criterion (-S, --select_r): Most conserved w/in 50 bases\n",
      "\t\t\tNote: Remaining conflicts will be randomly resolved.\n",
      "\t\tPrinting targets to file...\n",
      "\n",
      "\t\t### Results: 12 targets passed filtering! ###\n",
      "\t\t### Runtime: 0:00:01 (1.719325 seconds) ###\n",
      "\n",
      "\tStep 4: Bait discovery\n",
      "\t\tStep 4 parameters:\n",
      "\t\t\tBait length (-b, --blen): 90\n",
      "\t\t\tDesigning 2 baits per target with 20 maximum overlap\n",
      "\n",
      "\t\t### Results: 15 potential baits identified! ###\n",
      "\t\t### Runtime: 0:00:00 (0.097973 seconds) ###\n",
      "\n",
      "\tStep 5: Bait filtering\n",
      "\t\tFound 15 potential baits!\n",
      "\t\tApplying filters to 15 baits...\n",
      "\t\t\tFiltering criterion: Between 0.25 and 0.75 GC content\n",
      "\t\t\tFiltering criterion: BLAST exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_cp_only_9_45\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\t  --OVERRIDING BLAST DB: /moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.fa\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 24\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\t 15 passed filtering.\n",
      "\t\tFormatting for printing...\n",
      "\t\tFinal baits output to: /moto/eaton/users/slh2181/MrBait/bait_cp_only_9_45.fasta\n",
      "\n",
      "\t\t### Results: 15 baits output to file! ###\n",
      "\t\t### Runtime: 0:00:04 (4.844771 seconds) ###\n",
      "\n",
      "\t=======================================================================\n",
      "\t### Final output: /moto/eaton/users/slh2181/MrBait/bait_cp_only_9_45.fasta ###\n",
      "\t### Runtime: 1:16:26 (4586.895467 seconds) ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# for cpDNA\n",
    "#! mrbait -T 24 -L /moto/eaton/users/slh2181/MrBait/pooled-spp33_Hardmask_format.loci --cov 2 -l 100 --thresh 0.25 -Q 0.75 --mask 0.25 --max_mask 0.75 -b 90 --var_max 7 --numN 2 --numG 2 --mult_reg --min_mult 200 --dist_r 50 --flank_dist 50 --select_r cons --filter_r snp=1,25 -s calc=4,50 --filter_r gc=0.27,0.73 --filter_r pw=0.9,0.9 --print_loc --print_tr --out bait_only_cp --vthreads 24 --consens_maf 0.2 --fastadb /moto/eaton/users/slh2181/reference/hypochondriacus_cpDNA.fasta --filter_b blast_i=0.9,0.9 --nodust  \n",
    "#--dustMask   --filter_b pw=0.2,0.2 --filter_b gc=0.1,0.9 \n",
    "#--filter_r len=100,9999 \n",
    "\n",
    "! mrbait -T 24 --vthreads 24 -L /moto/eaton/users/slh2181/MrBait/pooled_9_45_outfiles/pooled_9_45.loci --cov 2 -l 100 --thresh 0.15 -Q 0.55 --mask 0.15 --max_mask 0.55 -b 90 --var_max 3 --numN 1 --numG 1 --mult_reg --min_mult 200 --dist_r 50 --flank_dist 50 --select_r cons --filter_r snp=1,20 -s calc=2,20 --filter_r pw=0.8,0.8 --filter_b pw=0.8,0.8 --filter_r gc=0.25,0.75 --filter_b gc=0.25,0.75 --print_loc --print_tr --out bait_cp_only_9_45 --consens_maf 0.2 --filter_r len=100,9999 --dustMask --fastadb /moto/eaton/users/slh2181/reference/hypochondriacus_cpDNA.fasta --filter_b blast_x=0.8,0.8 --blastx_fdb /moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.fa --filter_r blast_i=0.4,0.4 #--filter_b blast_a=0.2,0.2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### This was the older thing using the assembly from the old version of ipyrad"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.2.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /moto/eaton/users/slh2181/MrBait/bait_cp_only.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:06 (6.211091 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum alignment depth (-c): 2\n",
      "\t\t\tMinimum alignment length (-l): 100\n",
      "\t\t\tThreshold to call \"N\" or gap consensus base (-q): 0.15\n",
      "\t\t\tThreshold to retain masked base (-k): 0.15\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.55\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.55\n",
      "\t\t\tMinor allele frequency threshold (--consens_maf): 0.2\n",
      "\t\tLoading LOCI file: /moto/eaton/users/slh2181/MrBait/pooled-spp33_Hardmask_format.loci\n",
      "\t\t\tLoading alignments using 24 parallel processes.\n",
      "\t\t\tReading 49844 alignments... This may take a while.\n",
      "\t\tMasking consensus sequences using the DUST algorithm in VSEARCH... Done!\n",
      "\t\tFiltering loci... Done!\n",
      "\n",
      "\t\t### Results: 16394 loci passed filtering! ###\n",
      "\t\tPrinting locus catalog to file...\n",
      "\t\t### Runtime: 0:07:02 (422.262951 seconds) ###\n",
      "\n",
      "\tStep 2: Target Discovery\n",
      "\t\tStep 2 parameters:\n",
      "\t\t\tSliding window width (-b, --bait): 90\n",
      "\t\t\tSliding window shift distance (-w, --win_shift): 1\n",
      "\t\t\tVariable columns allowed per target (-v, --var_max): 3\n",
      "\t\t\tAmbiguities (N's) allowed per target (-n, --numN): 1\n",
      "\t\t\tGap characters allowed per target (-g, --numG): 1\n",
      "\t\t\tFlanking distance to parse (-d,--flank_dist): 50\n",
      "\t\tStarting sliding window target discovery of 16394 loci...\n",
      "\t\t\tFinding targets using 24 parallel processes...\n",
      "\n",
      "\t\t### Results: 12072 potential targets identified! ###\n",
      "\t\t### Runtime: 0:01:06 (66.666242 seconds) ###\n",
      "\n",
      "\tStep 3: Target Filtering and Selection\n",
      "\t\tStep 3 parameters:\n",
      "\t\t\tMultiple targets allowed per locus (-R, --mult_reg): True\n",
      "\t\t\tMinimum locus length for multiple targets (-m, --min_mult): 200\n",
      "\t\t\tMinimum distance between targets within locus (-D, --dist_r): 50\n",
      "\t\t\tFlanking distance for target selection (-d,--flank_dist): 50\n",
      "\t\tApplying filters to 12072 targets.\n",
      "\t\t\tFiltering criterion: Between 1 and 20 flanking SNPs\n",
      "\t\t\tFiltering criterion: Between 0.27 and 0.73 GC content\n",
      "\t\t\tFiltering criterion: Target length between 100 and 9999\n",
      "\t\t\tFiltering criterion: BLAST inclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_cp_only\n",
      "\t\t\t  --Percent identity: 0.4\n",
      "\t\t\t  --Query coverage: 0.4\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 24\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\tResolving conflicting targets... Found 2 needing resolution.\n",
      "\t\t\tSelection criterion (-S, --select_r): Most conserved w/in 50 bases\n",
      "\t\t\tNote: Remaining conflicts will be randomly resolved.\n",
      "\t\tPrinting targets to file...\n",
      "\n",
      "\t\t### Results: 14 targets passed filtering! ###\n",
      "\t\t### Runtime: 0:00:05 (5.031865 seconds) ###\n",
      "\n",
      "\tStep 4: Bait discovery\n",
      "\t\tStep 4 parameters:\n",
      "\t\t\tBait length (-b, --blen): 90\n",
      "\t\t\tDesigning 2 baits per target with 20 maximum overlap\n",
      "\n",
      "\t\t### Results: 17 potential baits identified! ###\n",
      "\t\t### Runtime: 0:00:00 (0.116069 seconds) ###\n",
      "\n",
      "\tStep 5: Bait filtering\n",
      "\t\tFound 17 potential baits!\n",
      "\t\tApplying filters to 17 baits...\n",
      "\t\t\tFiltering criterion: Between 0.27 and 0.73 GC content\n",
      "\t\t\tFiltering criterion: BLAST exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /moto/eaton/users/slh2181/MrBait/blastdb/bait_cp_only\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 24\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\t  --OVERRIDING BLAST DB: /moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.fa\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 24\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\t 16 passed filtering.\n",
      "\t\tFormatting for printing...\n",
      "\t\tFinal baits output to: /moto/eaton/users/slh2181/MrBait/bait_cp_only.fasta\n",
      "\n",
      "\t\t### Results: 16 baits output to file! ###\n",
      "\t\t### Runtime: 0:00:04 (4.906426 seconds) ###\n",
      "\n",
      "\t=======================================================================\n",
      "\t### Final output: /moto/eaton/users/slh2181/MrBait/bait_cp_only.fasta ###\n",
      "\t### Runtime: 0:08:25 (505.195572 seconds) ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# for cpDNA\n",
    "#! mrbait -T 24 -L /moto/eaton/users/slh2181/MrBait/pooled-spp33_Hardmask_format.loci --cov 2 -l 100 --thresh 0.25 -Q 0.75 --mask 0.25 --max_mask 0.75 -b 90 --var_max 7 --numN 2 --numG 2 --mult_reg --min_mult 200 --dist_r 50 --flank_dist 50 --select_r cons --filter_r snp=1,25 -s calc=4,50 --filter_r gc=0.27,0.73 --filter_r pw=0.9,0.9 --print_loc --print_tr --out bait_only_cp --vthreads 24 --consens_maf 0.2 --fastadb /moto/eaton/users/slh2181/reference/hypochondriacus_cpDNA.fasta --filter_b blast_i=0.9,0.9 --nodust  \n",
    "#--dustMask   --filter_b pw=0.2,0.2 --filter_b gc=0.1,0.9 \n",
    "#--filter_r len=100,9999 \n",
    "\n",
    "! mrbait -T 24 --vthreads 24 -L /moto/eaton/users/slh2181/MrBait/pooled-spp33_Hardmask_format.loci --cov 2 -l 100 --thresh 0.15 -Q 0.55 --mask 0.15 --max_mask 0.55 -b 90 --var_max 3 --numN 1 --numG 1 --mult_reg --min_mult 200 --dist_r 50 --flank_dist 50 --select_r cons --filter_r snp=1,20 -s calc=2,20 --filter_r pw=0.8,0.8 --filter_b pw=0.8,0.8 --filter_r gc=0.27,0.73 --filter_b gc=0.27,0.73 --print_loc --print_tr --out bait_cp_only --consens_maf 0.2 --filter_r len=100,9999 --dustMask --fastadb /moto/eaton/users/slh2181/reference/hypochondriacus_cpDNA.fasta --filter_b blast_x=0.8,0.8 --blastx_fdb /moto/eaton/users/slh2181/reference/Ahypochondriacus_459_v2.0.fa --filter_r blast_i=0.4,0.4 #--filter_b blast_a=0.2,0.2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Redoing with looser GC content..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.2.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /home/sandra/Files/MrBait/bait_no_cp_18jun2020.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:00 (0.012478 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum alignment depth (-c): 5\n",
      "\t\t\tMinimum alignment length (-l): 100\n",
      "\t\t\tThreshold to call \"N\" or gap consensus base (-q): 0.15\n",
      "\t\t\tThreshold to retain masked base (-k): 0.15\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.55\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.55\n",
      "\t\t\tMinor allele frequency threshold (--consens_maf): 0.2\n",
      "\t\tLoading LOCI file: /home/sandra/Files/MrBait/pooled_9_54_outfiles/pooled_9_54.loci\n",
      "\t\t\tLoading alignments using 70 parallel processes.\n",
      "\t\t\tReading 37316 alignments... This may take a while.\n",
      "\t\tMasking consensus sequences using the DUST algorithm in VSEARCH... Done!\n",
      "\t\tFiltering loci... Done!\n",
      "\n",
      "\t\t### Results: 8174 loci passed filtering! ###\n",
      "\t\tPrinting locus catalog to file...\n",
      "\t\t### Runtime: 0:58:16 (3496.326969 seconds) ###\n",
      "\n",
      "\tStep 2: Target Discovery\n",
      "\t\tStep 2 parameters:\n",
      "\t\t\tSliding window width (-b, --bait): 90\n",
      "\t\t\tSliding window shift distance (-w, --win_shift): 1\n",
      "\t\t\tVariable columns allowed per target (-v, --var_max): 3\n",
      "\t\t\tAmbiguities (N's) allowed per target (-n, --numN): 1\n",
      "\t\t\tGap characters allowed per target (-g, --numG): 1\n",
      "\t\t\tFlanking distance to parse (-d,--flank_dist): 50\n",
      "\t\tStarting sliding window target discovery of 8174 loci...\n",
      "\t\t\tFinding targets using 70 parallel processes...\n",
      "\n",
      "\t\t### Results: 6555 potential targets identified! ###\n",
      "\t\t### Runtime: 0:00:17 (17.098981 seconds) ###\n",
      "\n",
      "\tStep 3: Target Filtering and Selection\n",
      "\t\tStep 3 parameters:\n",
      "\t\t\tMultiple targets allowed per locus (-R, --mult_reg): True\n",
      "\t\t\tMinimum locus length for multiple targets (-m, --min_mult): 200\n",
      "\t\t\tMinimum distance between targets within locus (-D, --dist_r): 30\n",
      "\t\t\tFlanking distance for target selection (-d,--flank_dist): 50\n",
      "\t\tApplying filters to 6555 targets.\n",
      "\t\t\tFiltering criterion: Between 1 and 15 flanking SNPs\n",
      "\t\t\tFiltering criterion: Between 0.2 and 0.8 GC content\n",
      "\t\t\tFiltering criterion: Target length between 125 and 9999\n",
      "\t\t\tFiltering criterion: BLAST inclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /home/sandra/Files/MrBait/blastdb/bait_no_cp_18jun2020\n",
      "\t\t\t  --Percent identity: 0.5\n",
      "\t\t\t  --Query coverage: 0.5\n",
      "\t\t\t  --N threads: 70\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 70\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\tResolving edges...\n",
      "\t\t\t  --Resolve by graph: True\n",
      "\t\t\t  --MIS method: Weighted\n",
      "\t\t\t  --Weights: Maximum variation\n",
      "\t\tResolving conflicting targets... Found 280 needing resolution.\n",
      "\t\t\tSelection criterion (-S, --select_r): Most conserved w/in 50 bases\n",
      "\t\t\tNote: Remaining conflicts will be randomly resolved.\n",
      "\t\tPrinting targets to file...\n",
      "\n",
      "\t\t### Results: 2493 targets passed filtering! ###\n",
      "\t\t### Runtime: 0:00:23 (23.065870 seconds) ###\n",
      "\n",
      "\tStep 4: Bait discovery\n",
      "\t\tStep 4 parameters:\n",
      "\t\t\tBait length (-b, --blen): 90\n",
      "\t\t\tTiling baits with 45 base overlap\n",
      "\n",
      "\t\t### Results: 5169 potential baits identified! ###\n",
      "\t\t### Runtime: 0:00:11 (11.752636 seconds) ###\n",
      "\n",
      "\tStep 5: Bait filtering\n",
      "\t\tFound 5169 potential baits!\n",
      "\t\tApplying filters to 5169 baits...\n",
      "\t\t\tFiltering criterion: Between 0.2 and 0.8 GC content\n",
      "\t\t\tFiltering criterion: BLAST exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /home/sandra/Files/MrBait/blastdb/bait_no_cp_18jun2020\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 70\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\t  --OVERRIDING BLAST DB: /home/sandra/Files/MrBait/hypochondriacus_cpDNA.fasta\n",
      "\t\t\tFiltering criterion: BLAST ambiguous map exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /home/sandra/Files/MrBait/blastdb/bait_no_cp_18jun2020\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 70\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 70\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\tResolving edges...\n",
      "\t\t\t  --Resolve by graph: True\n",
      "\t\t\t  --MIS method: Weighted\n",
      "\t\t\t  --Weights: Maximum variation\n",
      "\t\t\t 5123 passed filtering.\n",
      "\t\tFormatting for printing...\n",
      "\t\tFinal baits output to: /home/sandra/Files/MrBait/bait_no_cp_18jun2020.fasta\n",
      "\n",
      "\t\t### Results: 5123 baits output to file! ###\n",
      "\t\t### Runtime: 0:00:04 (4.190004 seconds) ###\n",
      "\n",
      "\t=======================================================================\n",
      "\t### Final output: /home/sandra/Files/MrBait/bait_no_cp_18jun2020.fasta ###\n",
      "\t### Runtime: 0:59:12 (3552.447983 seconds) ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#nuclear DNA\n",
    "! mrbait -T 70 --vthreads 70 -L /home/sandra/Files/MrBait/pooled_9_54_outfiles/pooled_9_54.loci --cov 5 -l 100 --thresh 0.15 -Q 0.55 --mask 0.15 --max_mask 0.55 -b 90 --var_max 3 --numN 1 --numG 1 --mult_reg --min_mult 200 --dist_r 30 --flank_dist 50 --select_r cons --filter_r snp=1,15 -s tile=45 --filter_r pw=0.8,0.8 --filter_b pw=0.8,0.8 --filter_r gc=0.2,0.8 --filter_b gc=0.2,0.8 --print_loc --print_tr --out bait_no_cp_18jun2020 --consens_maf 0.2 --filter_r len=125,9999 --dustMask --blastx_fdb /home/sandra/Files/MrBait/hypochondriacus_cpDNA.fasta --filter_b blast_x=0.8,0.8 --fastadb /home/sandra/Files/MrBait/Ahypochondriacus_459_v2.0.fa --filter_r blast_i=0.5,0.5 --filter_b blast_a=0.8,0.8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.2.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /pinky/sandra/MrBait/bait_no_cp_29jun2020.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:00 (0.264752 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum alignment depth (-c): 5\n",
      "\t\t\tMinimum alignment length (-l): 100\n",
      "\t\t\tThreshold to call \"N\" or gap consensus base (-q): 0.15\n",
      "\t\t\tThreshold to retain masked base (-k): 0.15\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.55\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.55\n",
      "\t\t\tMinor allele frequency threshold (--consens_maf): 0.2\n",
      "\t\tLoading LOCI file: /pinky/sandra/MrBait/pooled_9_54_outfiles/pooled_9_54.loci\n",
      "\t\t\tLoading alignments using 55 parallel processes.\n",
      "\t\t\tReading 37316 alignments... This may take a while.\n",
      "\t\tMasking consensus sequences using the DUST algorithm in VSEARCH... Done!\n",
      "\t\tFiltering loci... Done!\n",
      "\n",
      "\t\t### Results: 8177 loci passed filtering! ###\n",
      "\t\tPrinting locus catalog to file...\n",
      "\t\t### Runtime: 1:25:02 (5102.278076 seconds) ###\n",
      "\n",
      "\tStep 2: Target Discovery\n",
      "\t\tStep 2 parameters:\n",
      "\t\t\tSliding window width (-b, --bait): 90\n",
      "\t\t\tSliding window shift distance (-w, --win_shift): 1\n",
      "\t\t\tVariable columns allowed per target (-v, --var_max): 3\n",
      "\t\t\tAmbiguities (N's) allowed per target (-n, --numN): 1\n",
      "\t\t\tGap characters allowed per target (-g, --numG): 1\n",
      "\t\t\tFlanking distance to parse (-d,--flank_dist): 50\n",
      "\t\tStarting sliding window target discovery of 8177 loci...\n",
      "\t\t\tFinding targets using 55 parallel processes...\n",
      "\n",
      "\t\t### Results: 6557 potential targets identified! ###\n",
      "\t\t### Runtime: 0:03:02 (182.811610 seconds) ###\n",
      "\n",
      "\tStep 3: Target Filtering and Selection\n",
      "\t\tStep 3 parameters:\n",
      "\t\t\tMultiple targets allowed per locus (-R, --mult_reg): True\n",
      "\t\t\tMinimum locus length for multiple targets (-m, --min_mult): 200\n",
      "\t\t\tMinimum distance between targets within locus (-D, --dist_r): 30\n",
      "\t\t\tFlanking distance for target selection (-d,--flank_dist): 50\n",
      "\t\tApplying filters to 6557 targets.\n",
      "\t\t\tFiltering criterion: Between 1 and 15 flanking SNPs\n",
      "\t\t\tFiltering criterion: Between 0.2 and 0.8 GC content\n",
      "\t\t\tFiltering criterion: Target length between 125 and 9999\n",
      "\t\t\tFiltering criterion: BLAST inclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_no_cp_29jun2020\n",
      "\t\t\t  --Percent identity: 0.5\n",
      "\t\t\t  --Query coverage: 0.5\n",
      "\t\t\t  --N threads: 55\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 55\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\tResolving edges...\n",
      "\t\t\t  --Resolve by graph: True\n",
      "\t\t\t  --MIS method: Weighted\n",
      "\t\t\t  --Weights: Maximum variation\n",
      "\t\tResolving conflicting targets... Found 279 needing resolution.\n",
      "\t\t\tSelection criterion (-S, --select_r): Most conserved w/in 50 bases\n",
      "\t\t\tNote: Remaining conflicts will be randomly resolved.\n",
      "\t\tPrinting targets to file...\n",
      "\n",
      "\t\t### Results: 2492 targets passed filtering! ###\n",
      "\t\t### Runtime: 0:00:25 (25.454655 seconds) ###\n",
      "\n",
      "\tStep 4: Bait discovery\n",
      "\t\tStep 4 parameters:\n",
      "\t\t\tBait length (-b, --blen): 90\n",
      "\t\t\tTiling baits with 45 base overlap\n",
      "\n",
      "\t\t### Results: 5171 potential baits identified! ###\n",
      "\t\t### Runtime: 0:02:20 (140.096931 seconds) ###\n",
      "\n",
      "\tStep 5: Bait filtering\n",
      "\t\tFound 5171 potential baits!\n",
      "\t\tApplying filters to 5171 baits...\n",
      "\t\t\tFiltering criterion: Between 0.2 and 0.8 GC content\n",
      "\t\t\tFiltering criterion: BLAST exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_no_cp_29jun2020\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 55\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\t  --OVERRIDING BLAST DB: /pinky/sandra/MrBait/hypochondriacus_cpDNA.fasta\n",
      "\t\t\tFiltering criterion: BLAST ambiguous map exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_no_cp_29jun2020\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 55\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 55\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\tResolving edges...\n",
      "\t\t\t  --Resolve by graph: True\n",
      "\t\t\t  --MIS method: Weighted\n",
      "\t\t\t  --Weights: Maximum variation\n",
      "\t\t\t 5124 passed filtering.\n",
      "\t\tFormatting for printing...\n",
      "\t\tFinal baits output to: /pinky/sandra/MrBait/bait_no_cp_29jun2020.fasta\n",
      "\n",
      "\t\t### Results: 5124 baits output to file! ###\n",
      "\t\t### Runtime: 0:00:04 (4.804662 seconds) ###\n",
      "\n",
      "\t=======================================================================\n",
      "\t### Final output: /pinky/sandra/MrBait/bait_no_cp_29jun2020.fasta ###\n",
      "\t### Runtime: 1:30:55 (5455.711621 seconds) ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#nuclear DNA ** This one was sent to Brian\n",
    "! mrbait -T 55 --vthreads 55 -L /pinky/sandra/MrBait/pooled_9_54_outfiles/pooled_9_54.loci --cov 5 -l 100 --thresh 0.15 -Q 0.55 --mask 0.15 --max_mask 0.55 -b 90 --var_max 3 --numN 1 --numG 1 --mult_reg --min_mult 200 --dist_r 30 --flank_dist 50 --select_r cons --filter_r snp=1,15 -s tile=45 --filter_r pw=0.8,0.8 --filter_b pw=0.8,0.8 --filter_r gc=0.2,0.8 --filter_b gc=0.2,0.8 --print_loc --print_tr --out bait_no_cp_29jun2020 --consens_maf 0.2 --filter_r len=125,9999 --dustMask --blastx_fdb /pinky/sandra/MrBait/hypochondriacus_cpDNA.fasta --filter_b blast_x=0.8,0.8 --fastadb /pinky/sandra/MrBait/Ahypochondriacus_459_v2.0.fa --filter_r blast_i=0.5,0.5 --filter_b blast_a=0.8,0.8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.2.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /pinky/sandra/MrBait/bait_cp_26jun2020.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:00 (0.659766 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum alignment depth (-c): 2\n",
      "\t\t\tMinimum alignment length (-l): 100\n",
      "\t\t\tThreshold to call \"N\" or gap consensus base (-q): 0.15\n",
      "\t\t\tThreshold to retain masked base (-k): 0.15\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.55\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.55\n",
      "\t\t\tMinor allele frequency threshold (--consens_maf): 0.3\n",
      "\t\tLoading LOCI file: /pinky/sandra/MrBait/pooled_9_54_outfiles/pooled_9_54.loci\n",
      "\t\t\tLoading alignments using 70 parallel processes.\n",
      "\t\t\tReading 37316 alignments... This may take a while.\n",
      "\t\tMasking consensus sequences using the DUST algorithm in VSEARCH... Done!\n",
      "\t\tFiltering loci... Done!\n",
      "\n",
      "\t\t### Results: 2209 loci passed filtering! ###\n",
      "\t\tPrinting locus catalog to file...\n",
      "\t\t### Runtime: 0:15:01 (901.669865 seconds) ###\n",
      "\n",
      "\tStep 2: Target Discovery\n",
      "\t\tStep 2 parameters:\n",
      "\t\t\tSliding window width (-b, --bait): 90\n",
      "\t\t\tSliding window shift distance (-w, --win_shift): 1\n",
      "\t\t\tVariable columns allowed per target (-v, --var_max): 3\n",
      "\t\t\tAmbiguities (N's) allowed per target (-n, --numN): 1\n",
      "\t\t\tGap characters allowed per target (-g, --numG): 1\n",
      "\t\t\tFlanking distance to parse (-d,--flank_dist): 50\n",
      "\t\tStarting sliding window target discovery of 2209 loci...\n",
      "\t\t\tFinding targets using 70 parallel processes...\n",
      "\n",
      "\t\t### Results: 2009 potential targets identified! ###\n",
      "\t\t### Runtime: 0:00:56 (56.583334 seconds) ###\n",
      "\n",
      "\tStep 3: Target Filtering and Selection\n",
      "\t\tStep 3 parameters:\n",
      "\t\t\tMultiple targets allowed per locus (-R, --mult_reg): True\n",
      "\t\t\tMinimum locus length for multiple targets (-m, --min_mult): 200\n",
      "\t\t\tMinimum distance between targets within locus (-D, --dist_r): 30\n",
      "\t\t\tFlanking distance for target selection (-d,--flank_dist): 50\n",
      "\t\tApplying filters to 2009 targets.\n",
      "\t\t\tFiltering criterion: Between 1 and 20 flanking SNPs\n",
      "\t\t\tFiltering criterion: Between 0.1 and 0.9 GC content\n",
      "\t\t\tFiltering criterion: Target length between 100 and 9999\n",
      "\t\t\tFiltering criterion: BLAST inclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_cp_26jun2020\n",
      "\t\t\t  --Percent identity: 0.4\n",
      "\t\t\t  --Query coverage: 0.4\n",
      "\t\t\t  --N threads: 70\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 70\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\tNo conflicting targets found. Skipping target selection.\n",
      "\t\tPrinting targets to file...\n",
      "\n",
      "\t\t### Results: 3 targets passed filtering! ###\n",
      "\t\t### Runtime: 0:00:01 (1.479093 seconds) ###\n",
      "\n",
      "\tStep 4: Bait discovery\n",
      "\t\tStep 4 parameters:\n",
      "\t\t\tBait length (-b, --blen): 90\n",
      "\t\t\tTiling baits with 45 base overlap\n",
      "\n",
      "\t\t### Results: 8 potential baits identified! ###\n",
      "\t\t### Runtime: 0:00:00 (0.332905 seconds) ###\n",
      "\n",
      "\tStep 5: Bait filtering\n",
      "\t\tFound 8 potential baits!\n",
      "\t\tApplying filters to 8 baits...\n",
      "\t\t\tFiltering criterion: Between 0.1 and 0.9 GC content\n",
      "\t\t\tFiltering criterion: BLAST exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_cp_26jun2020\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 70\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\t  --OVERRIDING BLAST DB: /pinky/sandra/MrBait/Ahypochondriacus_459_v2.0.fa\n",
      "\t\t\tFiltering criterion: BLAST ambiguous map exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_cp_26jun2020\n",
      "\t\t\t  --Percent identity: 0.6\n",
      "\t\t\t  --Query coverage: 0.6\n",
      "\t\t\t  --N threads: 70\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 70\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\t 8 passed filtering.\n",
      "\t\tFormatting for printing...\n",
      "\t\tFinal baits output to: /pinky/sandra/MrBait/bait_cp_26jun2020.fasta\n",
      "\n",
      "\t\t### Results: 8 baits output to file! ###\n",
      "\t\t### Runtime: 0:00:04 (4.667207 seconds) ###\n",
      "\n",
      "\t=======================================================================\n",
      "\t### Final output: /pinky/sandra/MrBait/bait_cp_26jun2020.fasta ###\n",
      "\t### Runtime: 0:16:05 (965.393050 seconds) ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#cpDNDA\n",
    "! mrbait -T 70 --vthreads 70 -L /pinky/sandra/MrBait/pooled_9_54_outfiles/pooled_9_54.loci --cov 2 -l 100 --thresh 0.15 -Q 0.55 --mask 0.15 --max_mask 0.55 -b 90 --var_max 3 --numN 1 --numG 1 --mult_reg --min_mult 200 --dist_r 30 --flank_dist 50 --select_r cons --filter_r snp=1,20 -s tile=45 --filter_r pw=0.8,0.8 --filter_b pw=0.8,0.8 --filter_r gc=0.1,0.9 --filter_b gc=0.1,0.9 --print_loc --print_tr --out bait_cp_26jun2020 --consens_maf 0.3 --filter_r len=100,9999 --dustMask --fastadb /pinky/sandra/MrBait/hypochondriacus_cpDNA.fasta --filter_b blast_x=0.8,0.8 --blastx_fdb /pinky/sandra/MrBait/Ahypochondriacus_459_v2.0.fa --filter_r blast_i=0.4,0.4 --filter_b blast_a=0.6,0.6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.2.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /pinky/sandra/MrBait/bait_cp_26jun2020.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:00 (0.783029 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum alignment depth (-c): 2\n",
      "\t\t\tMinimum alignment length (-l): 100\n",
      "\t\t\tThreshold to call \"N\" or gap consensus base (-q): 0.15\n",
      "\t\t\tThreshold to retain masked base (-k): 0.15\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.55\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.55\n",
      "\t\t\tMinor allele frequency threshold (--consens_maf): 0.2\n",
      "\t\tLoading LOCI file: /pinky/sandra/MrBait/pooled_9_54_outfiles/pooled_9_54.loci\n",
      "\t\t\tLoading alignments using 70 parallel processes.\n",
      "\t\t\tReading 37316 alignments... This may take a while.\n",
      "\t\tMasking consensus sequences using the DUST algorithm in VSEARCH... Done!\n",
      "\t\tFiltering loci... Done!\n",
      "\n",
      "\t\t### Results: 10130 loci passed filtering! ###\n",
      "\t\tPrinting locus catalog to file...\n",
      "\t\t### Runtime: 1:25:35 (5135.477362 seconds) ###\n",
      "\n",
      "\tStep 2: Target Discovery\n",
      "\t\tStep 2 parameters:\n",
      "\t\t\tSliding window width (-b, --bait): 90\n",
      "\t\t\tSliding window shift distance (-w, --win_shift): 1\n",
      "\t\t\tVariable columns allowed per target (-v, --var_max): 3\n",
      "\t\t\tAmbiguities (N's) allowed per target (-n, --numN): 1\n",
      "\t\t\tGap characters allowed per target (-g, --numG): 1\n",
      "\t\t\tFlanking distance to parse (-d,--flank_dist): 50\n",
      "\t\tStarting sliding window target discovery of 10130 loci...\n",
      "\t\t\tFinding targets using 70 parallel processes...\n",
      "\n",
      "\t\t### Results: 8021 potential targets identified! ###\n",
      "\t\t### Runtime: 0:03:47 (227.959862 seconds) ###\n",
      "\n",
      "\tStep 3: Target Filtering and Selection\n",
      "\t\tStep 3 parameters:\n",
      "\t\t\tMultiple targets allowed per locus (-R, --mult_reg): True\n",
      "\t\t\tMinimum locus length for multiple targets (-m, --min_mult): 200\n",
      "\t\t\tMinimum distance between targets within locus (-D, --dist_r): 30\n",
      "\t\t\tFlanking distance for target selection (-d,--flank_dist): 50\n",
      "\t\tApplying filters to 8021 targets.\n",
      "\t\t\tFiltering criterion: Between 1 and 20 flanking SNPs\n",
      "\t\t\tFiltering criterion: Between 0.1 and 0.9 GC content\n",
      "\t\t\tFiltering criterion: BLAST inclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_cp_26jun2020\n",
      "\t\t\t  --Percent identity: 0.4\n",
      "\t\t\t  --Query coverage: 0.4\n",
      "\t\t\t  --N threads: 70\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 70\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\tResolving conflicting targets... Found 2 needing resolution.\n",
      "\t\t\tSelection criterion (-S, --select_r): Most conserved w/in 50 bases\n",
      "\t\t\tNote: Remaining conflicts will be randomly resolved.\n",
      "\t\tPrinting targets to file...\n",
      "\n",
      "\t\t### Results: 13 targets passed filtering! ###\n",
      "\t\t### Runtime: 0:00:03 (3.681593 seconds) ###\n",
      "\n",
      "\tStep 4: Bait discovery\n",
      "\t\tStep 4 parameters:\n",
      "\t\t\tBait length (-b, --blen): 90\n",
      "\t\t\tTiling baits with 45 base overlap\n",
      "\n",
      "\t\t### Results: 24 potential baits identified! ###\n",
      "\t\t### Runtime: 0:00:00 (0.891498 seconds) ###\n",
      "\n",
      "\tStep 5: Bait filtering\n",
      "\t\tFound 24 potential baits!\n",
      "\t\tApplying filters to 24 baits...\n",
      "\t\t\tFiltering criterion: Between 0.1 and 0.9 GC content\n",
      "\t\t\tFiltering criterion: BLAST exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_cp_26jun2020\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 70\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\t  --OVERRIDING BLAST DB: /pinky/sandra/MrBait/Ahypochondriacus_459_v2.0.fa\n",
      "\t\t\tFiltering criterion: BLAST ambiguous map exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_cp_26jun2020\n",
      "\t\t\t  --Percent identity: 0.6\n",
      "\t\t\t  --Query coverage: 0.6\n",
      "\t\t\t  --N threads: 70\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 70\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\t 24 passed filtering.\n",
      "\t\tFormatting for printing...\n",
      "\t\tFinal baits output to: /pinky/sandra/MrBait/bait_cp_26jun2020.fasta\n",
      "\n",
      "\t\t### Results: 24 baits output to file! ###\n",
      "\t\t### Runtime: 0:00:05 (5.235870 seconds) ###\n",
      "\n",
      "\t=======================================================================\n",
      "\t### Final output: /pinky/sandra/MrBait/bait_cp_26jun2020.fasta ###\n",
      "\t### Runtime: 1:29:34 (5374.030239 seconds) ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#cpDNDA - no limit to locus length\n",
    "! mrbait -T 70 --vthreads 70 -L /pinky/sandra/MrBait/pooled_9_54_outfiles/pooled_9_54.loci --cov 2 -l 100 --thresh 0.15 -Q 0.55 --mask 0.15 --max_mask 0.55 -b 90 --var_max 3 --numN 1 --numG 1 --mult_reg --min_mult 200 --dist_r 30 --flank_dist 50 --select_r cons --filter_r snp=1,20 -s tile=45 --filter_r pw=0.8,0.8 --filter_b pw=0.8,0.8 --filter_r gc=0.1,0.9 --filter_b gc=0.1,0.9 --print_loc --print_tr --out bait_cp_26jun2020 --consens_maf 0.2 --dustMask --fastadb /pinky/sandra/MrBait/hypochondriacus_cpDNA.fasta --filter_b blast_x=0.8,0.8 --blastx_fdb /pinky/sandra/MrBait/Ahypochondriacus_459_v2.0.fa --filter_r blast_i=0.4,0.4 --filter_b blast_a=0.6,0.6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t=======================================================================\n",
      "\t    MrBait: Universal Probe Design for Targeted-Enrichment Methods\n",
      "\t=======================================================================\n",
      "\n",
      "\tVersion:  1.2.6\n",
      "\tAuthor:   Tyler K. Chafin\n",
      "\tContact:  tkchafin@uark.edu\n",
      "\tLicense:  GNU Public License v3.0\n",
      "\n",
      "\tReleases: https://github.com/tkchafin/mrbait/releases\n",
      "\n",
      "\tCitation: Chafin TK, Douglas MR, Douglas ME. 2018. MrBait: Universal\n",
      "\t\tidentification and design of targeted-enrichment capture probes.\n",
      "\t\tBioinformatics. https://doi.org/10.1093/bioinformatics/bty548\n",
      "\n",
      "\tSource code: https://github.com/tkchafin/mrbait\n",
      "\tDocumentation: https://mrbait.readthedocs.io/\n",
      "\tConda package: https://anaconda.org/tylerkchafin/mrbait\n",
      "\n",
      "\t=======================================================================\n",
      "\t\n",
      "\n",
      "\tParsing command line arguments...\n",
      "\tLoading SQLite Database...\n",
      "\t\tEstablishing database connection: /pinky/sandra/MrBait/bait_cp_26jun2020.sqlite\n",
      "\t\tInitializing empty tables.\n",
      "\n",
      "\t\t### Runtime: 0:00:01 (1.504638 seconds) ###\n",
      "\n",
      "\tStep 1: Loading Alignments\n",
      "\t\tStep 1 parameters:\n",
      "\t\t\tMinimum alignment depth (-c): 1\n",
      "\t\t\tMinimum alignment length (-l): 100\n",
      "\t\t\tThreshold to call \"N\" or gap consensus base (-q): 0.15\n",
      "\t\t\tThreshold to retain masked base (-k): 0.15\n",
      "\t\t\tMaximum allowed gap/N bases (-Q): 0.55\n",
      "\t\t\tMaximum allowed masked bases (-K): 0.55\n",
      "\t\t\tMinor allele frequency threshold (--consens_maf): 0.2\n",
      "\t\tLoading LOCI file: /pinky/sandra/MrBait/pooled_9_54_outfiles/pooled_9_54.loci\n",
      "\t\t\tLoading alignments using 80 parallel processes.\n",
      "\t\t\tReading 37316 alignments... This may take a while.\n",
      "\t\tMasking consensus sequences using the DUST algorithm in VSEARCH... Done!\n",
      "\t\tFiltering loci... Done!\n",
      "\n",
      "\t\t### Results: 10130 loci passed filtering! ###\n",
      "\t\tPrinting locus catalog to file...\n",
      "\t\t### Runtime: 1:24:19 (5059.151806 seconds) ###\n",
      "\n",
      "\tStep 2: Target Discovery\n",
      "\t\tStep 2 parameters:\n",
      "\t\t\tSliding window width (-b, --bait): 90\n",
      "\t\t\tSliding window shift distance (-w, --win_shift): 1\n",
      "\t\t\tVariable columns allowed per target (-v, --var_max): 3\n",
      "\t\t\tAmbiguities (N's) allowed per target (-n, --numN): 1\n",
      "\t\t\tGap characters allowed per target (-g, --numG): 1\n",
      "\t\t\tFlanking distance to parse (-d,--flank_dist): 50\n",
      "\t\tStarting sliding window target discovery of 10130 loci...\n",
      "\t\t\tFinding targets using 80 parallel processes...\n",
      "\n",
      "\t\t### Results: 8028 potential targets identified! ###\n",
      "\t\t### Runtime: 0:03:46 (226.390250 seconds) ###\n",
      "\n",
      "\tStep 3: Target Filtering and Selection\n",
      "\t\tStep 3 parameters:\n",
      "\t\t\tMultiple targets allowed per locus (-R, --mult_reg): True\n",
      "\t\t\tMinimum locus length for multiple targets (-m, --min_mult): 200\n",
      "\t\t\tMinimum distance between targets within locus (-D, --dist_r): 30\n",
      "\t\t\tFlanking distance for target selection (-d,--flank_dist): 50\n",
      "\t\tApplying filters to 8028 targets.\n",
      "\t\t\tFiltering criterion: Between 1 and 20 flanking SNPs\n",
      "\t\t\tFiltering criterion: Between 0.1 and 0.9 GC content\n",
      "\t\t\tFiltering criterion: BLAST inclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_cp_26jun2020\n",
      "\t\t\t  --Percent identity: 0.4\n",
      "\t\t\t  --Query coverage: 0.4\n",
      "\t\t\t  --N threads: 80\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 80\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\tResolving conflicting targets... Found 2 needing resolution.\n",
      "\t\t\tSelection criterion (-S, --select_r): Most conserved w/in 50 bases\n",
      "\t\t\tNote: Remaining conflicts will be randomly resolved.\n",
      "\t\tPrinting targets to file...\n",
      "\n",
      "\t\t### Results: 13 targets passed filtering! ###\n",
      "\t\t### Runtime: 0:00:03 (3.630460 seconds) ###\n",
      "\n",
      "\tStep 4: Bait discovery\n",
      "\t\tStep 4 parameters:\n",
      "\t\t\tBait length (-b, --blen): 90\n",
      "\t\t\tTiling baits with 45 base overlap\n",
      "\n",
      "\t\t### Results: 24 potential baits identified! ###\n",
      "\t\t### Runtime: 0:00:01 (1.043731 seconds) ###\n",
      "\n",
      "\tStep 5: Bait filtering\n",
      "\t\tFound 24 potential baits!\n",
      "\t\tApplying filters to 24 baits...\n",
      "\t\t\tFiltering criterion: Between 0.1 and 0.9 GC content\n",
      "\t\t\tFiltering criterion: BLAST exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_cp_26jun2020\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --N threads: 80\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\t  --OVERRIDING BLAST DB: /pinky/sandra/MrBait/Ahypochondriacus_459_v2.0.fa\n",
      "\t\t\tFiltering criterion: BLAST ambiguous map exclusion\n",
      "\t\t\t  --blastn path: blastn\n",
      "\t\t\t  --Database: /pinky/sandra/MrBait/blastdb/bait_cp_26jun2020\n",
      "\t\t\t  --Percent identity: 0.6\n",
      "\t\t\t  --Query coverage: 0.6\n",
      "\t\t\t  --N threads: 80\n",
      "\t\t\t  --Word size 11\n",
      "\t\t\t  --Gap open penalty: 5\n",
      "\t\t\t  --Gap extend: 2\n",
      "\t\t\t  --E-value cutoff: 1e-06\n",
      "\t\t\t  --Maximum returned hits: 10000\n",
      "\t\t\t  --DUST: True\n",
      "\t\t\t  --Method: blastn\n",
      "\t\t\tFiltering criterion: Pairwise alignment\n",
      "\t\t\t  --VSEARCH executable: vsearch\n",
      "\t\t\t  --VSEARCH threads: 80\n",
      "\t\t\t  --Percent identity: 0.8\n",
      "\t\t\t  --Query coverage: 0.8\n",
      "\t\t\t  --Masking method: dust\n",
      "\t\t\t 24 passed filtering.\n",
      "\t\tFormatting for printing...\n",
      "\t\tFinal baits output to: /pinky/sandra/MrBait/bait_cp_26jun2020.fasta\n",
      "\n",
      "\t\t### Results: 24 baits output to file! ###\n",
      "\t\t### Runtime: 0:00:05 (5.102324 seconds) ###\n",
      "\n",
      "\t=======================================================================\n",
      "\t### Final output: /pinky/sandra/MrBait/bait_cp_26jun2020.fasta ###\n",
      "\t### Runtime: 1:28:16 (5296.824230 seconds) ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#cpDNDA - no limit to locus length\n",
    "# ** This one was sent to Brian\n",
    "! mrbait -T 80 --vthreads 80 -L /pinky/sandra/MrBait/pooled_9_54_outfiles/pooled_9_54.loci --cov 1 -l 100 --thresh 0.15 -Q 0.55 --mask 0.15 --max_mask 0.55 -b 90 --var_max 3 --numN 1 --numG 1 --mult_reg --min_mult 200 --dist_r 30 --flank_dist 50 --select_r cons --filter_r snp=1,20 -s tile=45 --filter_r pw=0.8,0.8 --filter_b pw=0.8,0.8 --filter_r gc=0.1,0.9 --filter_b gc=0.1,0.9 --print_loc --print_tr --out bait_cp_26jun2020 --consens_maf 0.2 --dustMask --fastadb /pinky/sandra/MrBait/hypochondriacus_cpDNA.fasta --filter_b blast_x=0.8,0.8 --blastx_fdb /pinky/sandra/MrBait/Ahypochondriacus_459_v2.0.fa --filter_r blast_i=0.4,0.4 --filter_b blast_a=0.6,0.6"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
